%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}


\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{pymemdyn}
\date{Mar 20, 2023}
\release{1.6.1}
\author{H. Gutierrez de Teran \and X. Bello \and M. Esguerra \and R. L. van den Broek \and R.V. KÃ¼pper}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxstepscope


\chapter{Modules}
\label{\detokenize{modules:modules}}\label{\detokenize{modules::doc}}
\sphinxstepscope


\section{Pymemdyn module}
\label{\detokenize{pymemdyn:pymemdyn-module}}\label{\detokenize{pymemdyn::doc}}
\sphinxstepscope


\section{Protein module}
\label{\detokenize{protein:module-protein}}\label{\detokenize{protein:protein-module}}\label{\detokenize{protein::doc}}\index{module@\spxentry{module}!protein@\spxentry{protein}}\index{protein@\spxentry{protein}!module@\spxentry{module}}
\sphinxAtStartPar
This module handles the protein and all submitted molecules around it.
\index{ProteinComplex (class in protein)@\spxentry{ProteinComplex}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{ProteinComplex}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{setMonomer() (protein.ProteinComplex method)@\spxentry{setMonomer()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setMonomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setMonomer}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the monomer object.

\end{fulllineitems}

\index{getMonomer() (protein.ProteinComplex method)@\spxentry{getMonomer()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getMonomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getMonomer}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setLigand() (protein.ProteinComplex method)@\spxentry{setLigand()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setLigand}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setLigand}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the ligand object

\end{fulllineitems}

\index{getLigand() (protein.ProteinComplex method)@\spxentry{getLigand()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getLigand}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getLigand}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setWaters() (protein.ProteinComplex method)@\spxentry{setWaters()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setWaters}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the crystal waters object

\end{fulllineitems}

\index{getWaters() (protein.ProteinComplex method)@\spxentry{getWaters()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getWaters}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setIons() (protein.ProteinComplex method)@\spxentry{setIons()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setIons}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the ions object

\end{fulllineitems}

\index{getIons() (protein.ProteinComplex method)@\spxentry{getIons()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getIons}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setCho() (protein.ProteinComplex method)@\spxentry{setCho()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setCho}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setCho}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the cholesterol object

\end{fulllineitems}

\index{getCho() (protein.ProteinComplex method)@\spxentry{getCho()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getCho}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getCho}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setAlosteric() (protein.ProteinComplex method)@\spxentry{setAlosteric()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setAlosteric}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setAlosteric}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the allosteric object

\end{fulllineitems}

\index{getAlosteric() (protein.ProteinComplex method)@\spxentry{getAlosteric()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getAlosteric}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getAlosteric}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_nanom() (protein.ProteinComplex method)@\spxentry{set\_nanom()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.set_nanom}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_nanom}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Convert dimension measurements to nanometers for GROMACS

\end{fulllineitems}


\end{fulllineitems}

\index{Protein (class in protein)@\spxentry{Protein}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Protein}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{check\_number\_of\_chains() (protein.Protein method)@\spxentry{check\_number\_of\_chains()}\spxextra{protein.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Protein.check_number_of_chains}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_number\_of\_chains}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Determine if a PDB is a Monomer or a Dimer

\end{fulllineitems}


\end{fulllineitems}

\index{Monomer (class in protein)@\spxentry{Monomer}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Monomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Monomer}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{delete\_chain() (protein.Monomer method)@\spxentry{delete\_chain()}\spxextra{protein.Monomer method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Monomer.delete_chain}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_chain}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
PDBs which have a chain column mess up with pdb2gmx, creating
an unsuitable protein.itp file by naming the protein ie âProtein\_Aâ.
Here we remove the chain value

\sphinxAtStartPar
According to \sphinxurl{http://www.wwpdb.org/documentation/format33/sect9.html},
the chain value is in column 22

\end{fulllineitems}


\end{fulllineitems}

\index{Oligomer (class in protein)@\spxentry{Oligomer}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Oligomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Oligomer}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Monomer}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Monomer}}}}}
\index{delete\_chain() (protein.Oligomer method)@\spxentry{delete\_chain()}\spxextra{protein.Oligomer method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Oligomer.delete_chain}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{delete\_chain}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Overload the delete\_chain method from Monomer

\end{fulllineitems}


\end{fulllineitems}

\index{Sugar\_prep (class in protein)@\spxentry{Sugar\_prep}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Sugar_prep}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Sugar\_prep}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{create\_itp() (protein.Sugar\_prep method)@\spxentry{create\_itp()}\spxextra{protein.Sugar\_prep method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Sugar_prep.create_itp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_itp}}}{\emph{\DUrole{n}{pdbfile}\DUrole{p}{:}\DUrole{w}{  }\DUrole{n}{str}}, \emph{\DUrole{n}{charge}\DUrole{p}{:}\DUrole{w}{  }\DUrole{n}{int}}, \emph{\DUrole{n}{numberOfOptimizations}\DUrole{p}{:}\DUrole{w}{  }\DUrole{n}{int}}}{{ $\rightarrow$ None}}
\pysigstopsignatures
\sphinxAtStartPar
Call ligpargen to create gromacs itp file and corresponding openmm
pdb file. Note that original pdb file will be replaced by opnemm pdb
file.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pdbfile}} \textendash{} string containing local path to pdb of molecule. In commandline \sphinxhyphen{}i.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{charge}} \textendash{} interger charge of molecule. In commandline \sphinxhyphen{}c.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{numberOfOptimizations}} \textendash{} number of optimizations done by ligpargen. In cmdline \sphinxhyphen{}o.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\sphinxAtStartPar
Writes itp file and new pdf file to current dir. old pdb is saved in dir ligpargenInput. 
unneccessary ligpargen output is saved in dir ligpargenOutput.

\end{fulllineitems}

\index{lpg2pmd() (protein.Sugar\_prep method)@\spxentry{lpg2pmd()}\spxextra{protein.Sugar\_prep method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Sugar_prep.lpg2pmd}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{lpg2pmd}}}{\emph{\DUrole{n}{sugar}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Converts LigParGen structure files to PyMemDyn input files.

\sphinxAtStartPar
Original files are stored as something\_backup.pdb or something\_backup.itp.

\end{fulllineitems}


\end{fulllineitems}

\index{Compound (class in protein)@\spxentry{Compound}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Compound}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
This is a super\sphinxhyphen{}class to provide common functions to added compounds
\index{check\_files() (protein.Compound method)@\spxentry{check\_files()}\spxextra{protein.Compound method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound.check_files}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_files}}}{\emph{\DUrole{o}{*}\DUrole{n}{files}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Check if files passed as {\color{red}\bfseries{}*}args exist

\end{fulllineitems}


\end{fulllineitems}

\index{Ligand (class in protein)@\spxentry{Ligand}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ligand}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Ligand}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{check\_forces() (protein.Ligand method)@\spxentry{check\_forces()}\spxextra{protein.Ligand method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ligand.check_forces}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_forces}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
A force field must give a set of forces which match every atom in
the pdb file. This showed particularly important to the ligands, as they
may vary along a very broad range of atoms

\end{fulllineitems}


\end{fulllineitems}

\index{CrystalWaters (class in protein)@\spxentry{CrystalWaters}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{CrystalWaters}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{setWaters() (protein.CrystalWaters method)@\spxentry{setWaters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.setWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setWaters}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set crystal waters

\end{fulllineitems}

\index{getWaters() (protein.CrystalWaters method)@\spxentry{getWaters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.getWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getWaters}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal waters

\end{fulllineitems}

\index{number (protein.CrystalWaters property)@\spxentry{number}\spxextra{protein.CrystalWaters property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.number}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{number}}}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal waters

\end{fulllineitems}

\index{count\_waters() (protein.CrystalWaters method)@\spxentry{count\_waters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.count_waters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_waters}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Count and set the number of crystal waters in the pdb

\end{fulllineitems}


\end{fulllineitems}

\index{Ions (class in protein)@\spxentry{Ions}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Ions}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{setIons() (protein.Ions method)@\spxentry{setIons()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.setIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setIons}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the crystal ions

\end{fulllineitems}

\index{getIons() (protein.Ions method)@\spxentry{getIons()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.getIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getIons}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal ions

\end{fulllineitems}

\index{number (protein.Ions property)@\spxentry{number}\spxextra{protein.Ions property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.number}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{number}}}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal ions

\end{fulllineitems}

\index{count\_ions() (protein.Ions method)@\spxentry{count\_ions()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.count_ions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_ions}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Count and set the number of ions in the pdb

\end{fulllineitems}


\end{fulllineitems}

\index{Cholesterol (class in protein)@\spxentry{Cholesterol}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Cholesterol}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{setCho() (protein.Cholesterol method)@\spxentry{setCho()}\spxextra{protein.Cholesterol method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol.setCho}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setCho}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the crystal cholesterol

\end{fulllineitems}

\index{getCho() (protein.Cholesterol method)@\spxentry{getCho()}\spxextra{protein.Cholesterol method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol.getCho}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getCho}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal cholesterols

\end{fulllineitems}

\index{number (protein.Cholesterol property)@\spxentry{number}\spxextra{protein.Cholesterol property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol.number}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{number}}}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal cholesterols

\end{fulllineitems}

\index{check\_pdb() (protein.Cholesterol method)@\spxentry{check\_pdb()}\spxextra{protein.Cholesterol method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol.check_pdb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_pdb}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check the cholesterol file meets some standards

\end{fulllineitems}

\index{count\_cho() (protein.Cholesterol method)@\spxentry{count\_cho()}\spxextra{protein.Cholesterol method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Cholesterol.count_cho}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_cho}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Count and set the number of cho in the pdb

\end{fulllineitems}


\end{fulllineitems}

\index{Alosteric (class in protein)@\spxentry{Alosteric}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Alosteric}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Alosteric}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}

\sphinxAtStartPar
This is a compound that goes as a ligand but itâs placed in an allosteric
site rather than an orthosteric one.
\index{check\_pdb() (protein.Alosteric method)@\spxentry{check\_pdb()}\spxextra{protein.Alosteric method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Alosteric.check_pdb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_pdb}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check the allosteric file meets some standards

\end{fulllineitems}

\index{check\_itp() (protein.Alosteric method)@\spxentry{check\_itp()}\spxextra{protein.Alosteric method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Alosteric.check_itp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_itp}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check the force field is correct

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Membrane module}
\label{\detokenize{membrane:module-membrane}}\label{\detokenize{membrane:membrane-module}}\label{\detokenize{membrane::doc}}\index{module@\spxentry{module}!membrane@\spxentry{membrane}}\index{membrane@\spxentry{membrane}!module@\spxentry{module}}\index{Membrane (class in membrane)@\spxentry{Membrane}\spxextra{class in membrane}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{membrane:membrane.Membrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{membrane.}}\sphinxbfcode{\sphinxupquote{Membrane}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Set the characteristics of the membrane in the complex.
\index{set\_nanom() (membrane.Membrane method)@\spxentry{set\_nanom()}\spxextra{membrane.Membrane method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{membrane:membrane.Membrane.set_nanom}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_nanom}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Convert some measurements to nanometers to comply with GROMACS units.

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Bw4posres module}
\label{\detokenize{bw4posres:module-bw4posres}}\label{\detokenize{bw4posres:bw4posres-module}}\label{\detokenize{bw4posres::doc}}\index{module@\spxentry{module}!bw4posres@\spxentry{bw4posres}}\index{bw4posres@\spxentry{bw4posres}!module@\spxentry{module}}\begin{quote}

\sphinxAtStartPar
Date:        June 23, 2015
Email:       \sphinxhref{mailto:mauricio.esguerra@gmail.com}{mauricio.esguerra@gmail.com}

\sphinxAtStartPar
Description:
With this code we wish to do various task in one module:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Translate pdb to fasta without resorting to import Bio.

\item {} 
\sphinxAtStartPar
Align the translated fasta sequence to a Multiple Sequence Alignment (MSA)
and place Marks coming from a network of identified conserved
pair\sphinxhyphen{}distances of Venkatakrishnan et al.
clustalo \textendash{}profile1=GPCR\_inactive\_BWtags.aln \textendash{}profile2=mod1.fasta     \sphinxhyphen{}o withbwtags.aln \textendash{}outfmt=clustal \textendash{}wrap=1000 \textendash{}force \sphinxhyphen{}v \sphinxhyphen{}v \sphinxhyphen{}v

\item {} 
\sphinxAtStartPar
Translate Marks into properly identified residues in sequence. Notice that
this depends on a dictionary which uses the Ballesteros\sphinxhyphen{}Weinstein numbering.

\item {} 
\sphinxAtStartPar
From sequence ID. pull the atom\sphinxhyphen{}numbers of corresponding c\sphinxhyphen{}alphas
in the matched residues.

\end{enumerate}
\end{quote}


\bigskip\hrule\bigskip

\index{Run (class in bw4posres)@\spxentry{Run}\spxextra{class in bw4posres}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{bw4posres.}}\sphinxbfcode{\sphinxupquote{Run}}}{\emph{\DUrole{n}{pdb}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A pdb file is given as input to convert into one letter sequence
and then align to curated multiple sequence alignment and then
assign Ballesteros\sphinxhyphen{}Weinstein numbering to special positions.
\index{pdb2fas() (bw4posres.Run method)@\spxentry{pdb2fas()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.pdb2fas}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pdb2fas}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
From pdb file convert to fasta sequence format without the use of
dependencies such as BioPython. This pdb to fasta translator
checks for the existance of c\sphinxhyphen{}alpha residues and it is
based on their 3\sphinxhyphen{}letter sequence id.

\end{fulllineitems}

\index{clustalalign() (bw4posres.Run method)@\spxentry{clustalalign()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.clustalalign}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{clustalalign}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Align the produced fasta sequence with clustalw to assing
Ballesteros\sphinxhyphen{}Weinstein marks.

\end{fulllineitems}

\index{getcalphas() (bw4posres.Run method)@\spxentry{getcalphas()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.getcalphas}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getcalphas}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Pulls out the atom numbers of c\sphinxhyphen{}alpha atoms. Restraints are
placed on c\sphinxhyphen{}alpha atoms.

\end{fulllineitems}

\index{makedisre() (bw4posres.Run method)@\spxentry{makedisre()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.makedisre}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{makedisre}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a disre.itp file with atom\sphinxhyphen{}pair idâs to be restrained
using and NMR\sphinxhyphen{}style Heaviside function based on
Ballesteros\sphinxhyphen{}Weinstein tagging.

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Complex module}
\label{\detokenize{complex:module-complex}}\label{\detokenize{complex:complex-module}}\label{\detokenize{complex::doc}}\index{module@\spxentry{module}!complex@\spxentry{complex}}\index{complex@\spxentry{complex}!module@\spxentry{module}}\index{MembraneComplex (class in complex)@\spxentry{MembraneComplex}\spxextra{class in complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{complex.}}\sphinxbfcode{\sphinxupquote{MembraneComplex}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{setMembrane() (complex.MembraneComplex method)@\spxentry{setMembrane()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.setMembrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setMembrane}}}{\emph{\DUrole{n}{membrane}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set the membrane pdb file

\end{fulllineitems}

\index{getMembrane() (complex.MembraneComplex method)@\spxentry{getMembrane()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.getMembrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getMembrane}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setComplex() (complex.MembraneComplex method)@\spxentry{setComplex()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.setComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setComplex}}}{\emph{\DUrole{n}{complex}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set the complex object

\end{fulllineitems}

\index{getComplex() (complex.MembraneComplex method)@\spxentry{getComplex()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.getComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getComplex}}}{}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Queue module}
\label{\detokenize{queue:module-queue}}\label{\detokenize{queue:queue-module}}\label{\detokenize{queue::doc}}\index{module@\spxentry{module}!queue@\spxentry{queue}}\index{queue@\spxentry{queue}!module@\spxentry{module}}
\sphinxAtStartPar
A multi\sphinxhyphen{}producer, multi\sphinxhyphen{}consumer queue.
\index{Empty@\spxentry{Empty}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Empty}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Empty}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{Exception}}

\sphinxAtStartPar
Exception raised by Queue.get(block=0)/get\_nowait().

\end{fulllineitems}

\index{Full@\spxentry{Full}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Full}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Full}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{Exception}}

\sphinxAtStartPar
Exception raised by Queue.put(block=0)/put\_nowait().

\end{fulllineitems}

\index{Queue (class in queue)@\spxentry{Queue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Queue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Create a queue object with a given maximum size.

\sphinxAtStartPar
If maxsize is \textless{}= 0, the queue size is infinite.
\index{task\_done() (queue.Queue method)@\spxentry{task\_done()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.task_done}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{task\_done}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Indicate that a formerly enqueued task is complete.

\sphinxAtStartPar
Used by Queue consumer threads.  For each get() used to fetch a task,
a subsequent call to task\_done() tells the queue that the processing
on the task is complete.

\sphinxAtStartPar
If a join() is currently blocking, it will resume when all items
have been processed (meaning that a task\_done() call was received
for every item that had been put() into the queue).

\sphinxAtStartPar
Raises a ValueError if called more times than there were items
placed in the queue.

\end{fulllineitems}

\index{join() (queue.Queue method)@\spxentry{join()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.join}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{join}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Blocks until all items in the Queue have been gotten and processed.

\sphinxAtStartPar
The count of unfinished tasks goes up whenever an item is added to the
queue. The count goes down whenever a consumer thread calls task\_done()
to indicate the item was retrieved and all work on it is complete.

\sphinxAtStartPar
When the count of unfinished tasks drops to zero, join() unblocks.

\end{fulllineitems}

\index{qsize() (queue.Queue method)@\spxentry{qsize()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.qsize}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{qsize}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return the approximate size of the queue (not reliable!).

\end{fulllineitems}

\index{empty() (queue.Queue method)@\spxentry{empty()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.empty}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{empty}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return True if the queue is empty, False otherwise (not reliable!).

\sphinxAtStartPar
This method is likely to be removed at some point.  Use qsize() == 0
as a direct substitute, but be aware that either approach risks a race
condition where a queue can grow before the result of empty() or
qsize() can be used.

\sphinxAtStartPar
To create code that needs to wait for all queued tasks to be
completed, the preferred technique is to use the join() method.

\end{fulllineitems}

\index{full() (queue.Queue method)@\spxentry{full()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.full}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{full}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return True if the queue is full, False otherwise (not reliable!).

\sphinxAtStartPar
This method is likely to be removed at some point.  Use qsize() \textgreater{}= n
as a direct substitute, but be aware that either approach risks a race
condition where a queue can shrink before the result of full() or
qsize() can be used.

\end{fulllineitems}

\index{put() (queue.Queue method)@\spxentry{put()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.put}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put}}}{\emph{\DUrole{n}{item}}, \emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put an item into the queue.

\sphinxAtStartPar
If optional args âblockâ is true and âtimeoutâ is None (the default),
block if necessary until a free slot is available. If âtimeoutâ is
a non\sphinxhyphen{}negative number, it blocks at most âtimeoutâ seconds and raises
the Full exception if no free slot was available within that time.
Otherwise (âblockâ is false), put an item on the queue if a free slot
is immediately available, else raise the Full exception (âtimeoutâ
is ignored in that case).

\end{fulllineitems}

\index{get() (queue.Queue method)@\spxentry{get()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.get}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get}}}{\emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue.

\sphinxAtStartPar
If optional args âblockâ is true and âtimeoutâ is None (the default),
block if necessary until an item is available. If âtimeoutâ is
a non\sphinxhyphen{}negative number, it blocks at most âtimeoutâ seconds and raises
the Empty exception if no item was available within that time.
Otherwise (âblockâ is false), return an item if one is immediately
available, else raise the Empty exception (âtimeoutâ is ignored
in that case).

\end{fulllineitems}

\index{put\_nowait() (queue.Queue method)@\spxentry{put\_nowait()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.put_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put\_nowait}}}{\emph{\DUrole{n}{item}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put an item into the queue without blocking.

\sphinxAtStartPar
Only enqueue the item if a free slot is immediately available.
Otherwise raise the Full exception.

\end{fulllineitems}

\index{get\_nowait() (queue.Queue method)@\spxentry{get\_nowait()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.get_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nowait}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue without blocking.

\sphinxAtStartPar
Only get an item if one is immediately available. Otherwise
raise the Empty exception.

\end{fulllineitems}


\end{fulllineitems}

\index{PriorityQueue (class in queue)@\spxentry{PriorityQueue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.PriorityQueue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{PriorityQueue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{queue:queue.Queue}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Queue}}}}}

\sphinxAtStartPar
Variant of Queue that retrieves open entries in priority order (lowest first).

\sphinxAtStartPar
Entries are typically tuples of the form:  (priority number, data).

\end{fulllineitems}

\index{LifoQueue (class in queue)@\spxentry{LifoQueue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.LifoQueue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{LifoQueue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{queue:queue.Queue}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Queue}}}}}

\sphinxAtStartPar
Variant of Queue that retrieves most recently added entries first.

\end{fulllineitems}

\index{SimpleQueue (class in queue)@\spxentry{SimpleQueue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{SimpleQueue}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Simple, unbounded, reentrant FIFO queue.
\index{empty() (queue.SimpleQueue method)@\spxentry{empty()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.empty}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{empty}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return True if the queue is empty, False otherwise (not reliable!).

\end{fulllineitems}

\index{get() (queue.SimpleQueue method)@\spxentry{get()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.get}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get}}}{\emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue.

\sphinxAtStartPar
If optional args âblockâ is true and âtimeoutâ is None (the default),
block if necessary until an item is available. If âtimeoutâ is
a non\sphinxhyphen{}negative number, it blocks at most âtimeoutâ seconds and raises
the Empty exception if no item was available within that time.
Otherwise (âblockâ is false), return an item if one is immediately
available, else raise the Empty exception (âtimeoutâ is ignored
in that case).

\end{fulllineitems}

\index{get\_nowait() (queue.SimpleQueue method)@\spxentry{get\_nowait()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.get_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nowait}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue without blocking.

\sphinxAtStartPar
Only get an item if one is immediately available. Otherwise
raise the Empty exception.

\end{fulllineitems}

\index{put() (queue.SimpleQueue method)@\spxentry{put()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.put}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put}}}{\emph{\DUrole{n}{item}}, \emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put the item on the queue.

\sphinxAtStartPar
The optional âblockâ and âtimeoutâ arguments are ignored, as this method
never blocks.  They are provided for compatibility with the Queue class.

\end{fulllineitems}

\index{put\_nowait() (queue.SimpleQueue method)@\spxentry{put\_nowait()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.put_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put\_nowait}}}{\emph{\DUrole{n}{item}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put an item into the queue without blocking.

\sphinxAtStartPar
This is exactly equivalent to \sphinxtitleref{put(item)} and is only provided
for compatibility with the Queue class.

\end{fulllineitems}

\index{qsize() (queue.SimpleQueue method)@\spxentry{qsize()}\spxextra{queue.SimpleQueue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.SimpleQueue.qsize}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{qsize}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return the approximate size of the queue (not reliable!).

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Recipes module}
\label{\detokenize{recipes:module-recipes}}\label{\detokenize{recipes:recipes-module}}\label{\detokenize{recipes::doc}}\index{module@\spxentry{module}!recipes@\spxentry{recipes}}\index{recipes@\spxentry{recipes}!module@\spxentry{module}}
\sphinxAtStartPar
This module describes the commandline or python commands for all the 
phases of pymemdyn. It consists of:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Init

\item {} 
\sphinxAtStartPar
Minimization

\item {} 
\sphinxAtStartPar
Equilibration

\item {} 
\sphinxAtStartPar
Relaxation

\item {} 
\sphinxAtStartPar
Collecting results

\end{itemize}
\index{BasicInit (class in recipes)@\spxentry{BasicInit}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicInit}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicInit}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandInit (class in recipes)@\spxentry{LigandInit}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandInit}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandInit}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicInit}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicInit}}}}}

\end{fulllineitems}

\index{LigandAlostericInit (class in recipes)@\spxentry{LigandAlostericInit}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandAlostericInit}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandAlostericInit}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.LigandInit}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LigandInit}}}}}

\end{fulllineitems}

\index{BasicMinimization (class in recipes)@\spxentry{BasicMinimization}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicMinimization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicMinimization}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandMinimization (class in recipes)@\spxentry{LigandMinimization}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandMinimization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandMinimization}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicMinimization}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicMinimization}}}}}

\end{fulllineitems}

\index{LigandAlostericMinimization (class in recipes)@\spxentry{LigandAlostericMinimization}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandAlostericMinimization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandAlostericMinimization}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicMinimization}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicMinimization}}}}}

\end{fulllineitems}

\index{BasicEquilibration (class in recipes)@\spxentry{BasicEquilibration}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicEquilibration}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicEquilibration}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandEquilibration (class in recipes)@\spxentry{LigandEquilibration}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandEquilibration}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandEquilibration}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicEquilibration}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicEquilibration}}}}}

\end{fulllineitems}

\index{LigandAlostericEquilibration (class in recipes)@\spxentry{LigandAlostericEquilibration}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandAlostericEquilibration}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandAlostericEquilibration}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.LigandEquilibration}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LigandEquilibration}}}}}

\end{fulllineitems}

\index{BasicRelax (class in recipes)@\spxentry{BasicRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandRelax (class in recipes)@\spxentry{LigandRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicRelax}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicRelax}}}}}

\end{fulllineitems}

\index{LigandAlostericRelax (class in recipes)@\spxentry{LigandAlostericRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandAlostericRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandAlostericRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.LigandRelax}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{LigandRelax}}}}}

\end{fulllineitems}

\index{BasicCARelax (class in recipes)@\spxentry{BasicCARelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCARelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCARelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicBWRelax (class in recipes)@\spxentry{BasicBWRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicBWRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicBWRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicCollectResults (class in recipes)@\spxentry{BasicCollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicCACollectResults (class in recipes)@\spxentry{BasicCACollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCACollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCACollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicCollectResults}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicCollectResults}}}}}

\end{fulllineitems}

\index{BasicBWCollectResults (class in recipes)@\spxentry{BasicBWCollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicBWCollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicBWCollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicCollectResults}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicCollectResults}}}}}

\end{fulllineitems}


\sphinxstepscope


\section{Gromacs module}
\label{\detokenize{gromacs:module-gromacs}}\label{\detokenize{gromacs:gromacs-module}}\label{\detokenize{gromacs::doc}}\index{module@\spxentry{module}!gromacs@\spxentry{gromacs}}\index{gromacs@\spxentry{gromacs}!module@\spxentry{module}}\index{Gromacs (class in gromacs)@\spxentry{Gromacs}\spxextra{class in gromacs}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{gromacs.}}\sphinxbfcode{\sphinxupquote{Gromacs}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{set\_membrane\_complex() (gromacs.Gromacs method)@\spxentry{set\_membrane\_complex()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_membrane_complex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_membrane\_complex}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_membrane\_complex: Sets the monomer object

\end{fulllineitems}

\index{get\_membrane\_complex() (gromacs.Gromacs method)@\spxentry{get\_membrane\_complex()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_membrane_complex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_membrane\_complex}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{membrane\_complex (gromacs.Gromacs property)@\spxentry{membrane\_complex}\spxextra{gromacs.Gromacs property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.membrane_complex}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{membrane\_complex}}}
\pysigstopsignatures
\end{fulllineitems}

\index{count\_lipids() (gromacs.Gromacs method)@\spxentry{count\_lipids()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.count_lipids}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_lipids}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
count\_lipids: Counts the lipids in source and writes a target with N4 tags

\end{fulllineitems}

\index{get\_charge() (gromacs.Gromacs method)@\spxentry{get\_charge()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_charge}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_charge}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_charge: Gets the total charge of a system using gromacs grompp command

\end{fulllineitems}

\index{get\_ndx\_groups() (gromacs.Gromacs method)@\spxentry{get\_ndx\_groups()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_ndx_groups}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_ndx\_groups}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_ndx\_groups: Run make\_ndx and set the total number of groups found

\end{fulllineitems}

\index{get\_ndx\_sol() (gromacs.Gromacs method)@\spxentry{get\_ndx\_sol()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_ndx_sol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_ndx\_sol}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_ndx\_sol: Run make\_ndx and set the last number id for SOL found

\end{fulllineitems}

\index{make\_ndx() (gromacs.Gromacs method)@\spxentry{make\_ndx()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.make_ndx}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{make\_ndx}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
make\_ndx: Wraps the make\_ndx command tweaking the input to reflect the
characteristics of the complex

\end{fulllineitems}

\index{make\_topol\_lipids() (gromacs.Gromacs method)@\spxentry{make\_topol\_lipids()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.make_topol_lipids}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{make\_topol\_lipids}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
make\_topol\_lipids: Add lipid positions to topol.top

\end{fulllineitems}

\index{manual\_log() (gromacs.Gromacs method)@\spxentry{manual\_log()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.manual_log}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{manual\_log}}}{\emph{\DUrole{n}{command}}, \emph{\DUrole{n}{output}}}{}
\pysigstopsignatures
\sphinxAtStartPar
manual\_log: Redirect the output to file in command{[}âoptionsâ{]}{[}âlogâ{]}
Some commands canât be logged via flag, so one has to catch and
redirect stdout and stderr

\end{fulllineitems}

\index{relax() (gromacs.Gromacs method)@\spxentry{relax()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.relax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{relax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
relax: Relax a protein

\end{fulllineitems}

\index{run\_recipe() (gromacs.Gromacs method)@\spxentry{run\_recipe()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.run_recipe}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run\_recipe}}}{\emph{\DUrole{n}{debug}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
run\_recipe: Run recipe for the complex

\end{fulllineitems}

\index{select\_recipe() (gromacs.Gromacs method)@\spxentry{select\_recipe()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.select_recipe}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{select\_recipe}}}{\emph{\DUrole{n}{stage}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{debug}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
select\_recipe: Select the appropriate recipe for the complex

\end{fulllineitems}

\index{set\_box\_sizes() (gromacs.Gromacs method)@\spxentry{set\_box\_sizes()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_box_sizes}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_box\_sizes}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_box\_sizes: Set length values for different boxes

\end{fulllineitems}

\index{set\_chains() (gromacs.Gromacs method)@\spxentry{set\_chains()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_chains}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_chains}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_chains: Set the REAL points of a dimer after protonation

\end{fulllineitems}

\index{set\_grompp() (gromacs.Gromacs method)@\spxentry{set\_grompp()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_grompp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_grompp}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_grompp: Copy template files to working dir

\end{fulllineitems}

\index{set\_itp() (gromacs.Gromacs method)@\spxentry{set\_itp()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_itp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_itp}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_itp: Cut a top file to be usable later as itp

\end{fulllineitems}

\index{set\_options() (gromacs.Gromacs method)@\spxentry{set\_options()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_options}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_options}}}{\emph{\DUrole{n}{options}}, \emph{\DUrole{n}{breaks}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_options: Set break options from recipe

\end{fulllineitems}

\index{set\_popc() (gromacs.Gromacs method)@\spxentry{set\_popc()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_popc}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_popc}}}{\emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_popc: Create a pdb file only with the lipid bilayer (POP), no waters.
Set some measures on the fly (height of the bilayer)

\end{fulllineitems}

\index{set\_protein\_height() (gromacs.Gromacs method)@\spxentry{set\_protein\_height()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_protein_height}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_protein\_height}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_protein\_height: Get the z\sphinxhyphen{}axis center from a pdb file for membrane or
solvent alignment

\end{fulllineitems}

\index{set\_protein\_size() (gromacs.Gromacs method)@\spxentry{set\_protein\_size()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_protein_size}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_protein\_size}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_protein\_size: Get the protein maximum base width from a pdb file

\end{fulllineitems}

\index{set\_stage\_init() (gromacs.Gromacs method)@\spxentry{set\_stage\_init()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_stage_init}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_stage\_init}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_stage\_init: Copy a set of files from source to target dir

\end{fulllineitems}

\index{set\_steep() (gromacs.Gromacs method)@\spxentry{set\_steep()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_steep}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_steep}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_steep: Copy the template steep.mdp to target dir

\end{fulllineitems}

\index{set\_water() (gromacs.Gromacs method)@\spxentry{set\_water()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_water}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_water}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_water: Create a water layer for a box

\end{fulllineitems}


\end{fulllineitems}

\index{Wrapper (class in gromacs)@\spxentry{Wrapper}\spxextra{class in gromacs}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{gromacs.}}\sphinxbfcode{\sphinxupquote{Wrapper}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{generate\_command() (gromacs.Wrapper method)@\spxentry{generate\_command()}\spxextra{gromacs.Wrapper method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper.generate_command}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_command}}}{\emph{\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
generate\_command: Receive some variables in kwargs, generate
the appropriate command to be run. Return a set in the form of
a string âcommand \sphinxhyphen{}with flagsâ

\end{fulllineitems}

\index{run\_command() (gromacs.Wrapper method)@\spxentry{run\_command()}\spxextra{gromacs.Wrapper method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper.run_command}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run\_command}}}{\emph{\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
run\_command: Run a command that comes in kwargs in a subprocess, and
return the output as (output, errors)

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Groerrors module}
\label{\detokenize{groerrors:module-groerrors}}\label{\detokenize{groerrors:groerrors-module}}\label{\detokenize{groerrors::doc}}\index{module@\spxentry{module}!groerrors@\spxentry{groerrors}}\index{groerrors@\spxentry{groerrors}!module@\spxentry{module}}\index{GromacsError@\spxentry{GromacsError}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsError}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{GromacsError}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{BaseException}}

\end{fulllineitems}

\index{IOGromacsError@\spxentry{IOGromacsError}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.IOGromacsError}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{IOGromacsError}}}{\emph{\DUrole{n}{command}}, \emph{\DUrole{n}{explain}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{groerrors:groerrors.GromacsError}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{GromacsError}}}}}

\sphinxAtStartPar
Exception raised with âFile input/output errorâ message

\end{fulllineitems}

\index{GromacsMessages (class in groerrors)@\spxentry{GromacsMessages}\spxextra{class in groerrors}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{GromacsMessages}}}{\emph{\DUrole{n}{gro\_err}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{command}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Load an error message and split it along as many properties as
possible
\index{e (groerrors.GromacsMessages attribute)@\spxentry{e}\spxextra{groerrors.GromacsMessages attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages.e}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{e}}\sphinxbfcode{\sphinxupquote{\DUrole{w}{  }\DUrole{p}{=}\DUrole{w}{  }\{\textquotesingle{}Can not open file\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}, \textquotesingle{}File input/output error\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}, \textquotesingle{}srun: error: Unable to create job step\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}\}}}}
\pysigstopsignatures
\end{fulllineitems}

\index{check() (groerrors.GromacsMessages method)@\spxentry{check()}\spxextra{groerrors.GromacsMessages method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages.check}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check if the GROMACS error message has any of the known error
messages. Set the self.error to the value of the error

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Broker module}
\label{\detokenize{broker:module-broker}}\label{\detokenize{broker:broker-module}}\label{\detokenize{broker::doc}}\index{module@\spxentry{module}!broker@\spxentry{broker}}\index{broker@\spxentry{broker}!module@\spxentry{module}}
\sphinxAtStartPar
This is a lame broker (or message dispatcher). When Gromacs enters a run, 
it should choose a broker from here and dispatch messages through it.

\sphinxAtStartPar
Depending on the broker, the messages may be just printed or something else
\index{Printing (class in broker)@\spxentry{Printing}\spxextra{class in broker}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{broker:broker.Printing}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{broker.}}\sphinxbfcode{\sphinxupquote{Printing}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{dispatch() (broker.Printing method)@\spxentry{dispatch()}\spxextra{broker.Printing method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{broker:broker.Printing.dispatch}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dispatch}}}{\emph{\DUrole{n}{msg}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Simply print the msg passed

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Utils module}
\label{\detokenize{utils:module-utils}}\label{\detokenize{utils:utils-module}}\label{\detokenize{utils::doc}}\index{module@\spxentry{module}!utils@\spxentry{utils}}\index{utils@\spxentry{utils}!module@\spxentry{module}}\index{clean\_pdb() (in module utils)@\spxentry{clean\_pdb()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.clean_pdb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{clean\_pdb}}}{\emph{\DUrole{n}{src}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove incorrectly allocated atom identifiers in pdb file

\end{fulllineitems}

\index{clean\_topol() (in module utils)@\spxentry{clean\_topol()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.clean_topol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{clean\_topol}}}{\emph{\DUrole{n}{src}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Clean the src topol of path specifics, and paste results in target

\end{fulllineitems}

\index{concat() (in module utils)@\spxentry{concat()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.concat}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{concat}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Make a whole pdb file with all the pdb provided

\end{fulllineitems}

\index{getbw() (in module utils)@\spxentry{getbw()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.getbw}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{getbw}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Call the Ballesteros\sphinxhyphen{}Weistein based pair\sphinxhyphen{}distance restraint
module.

\end{fulllineitems}

\index{make\_cat() (in module utils)@\spxentry{make\_cat()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_cat}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_cat}}}{\emph{\DUrole{n}{dir1}}, \emph{\DUrole{n}{dir2}}, \emph{\DUrole{n}{name}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Very tight function to make a list of files to inject
in some GROMACS suite programs

\end{fulllineitems}

\index{make\_ffoplsaanb() (in module utils)@\spxentry{make\_ffoplsaanb()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_ffoplsaanb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_ffoplsaanb}}}{\emph{\DUrole{n}{complex}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Join all OPLS force fields needed to run the simulation

\end{fulllineitems}

\index{make\_topol() (in module utils)@\spxentry{make\_topol()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_topol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_topol}}}{\emph{\DUrole{n}{template\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}/home/rebecca/Documents/8STAGE/pymemdyn/templates\textquotesingle{}}}, \emph{\DUrole{n}{target\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{working\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{complex}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Make the topol starting from our topol.top template

\end{fulllineitems}

\index{tar\_out() (in module utils)@\spxentry{tar\_out()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.tar_out}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{tar\_out}}}{\emph{\DUrole{n}{src\_dir}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Tar everything in a src\_dir to the tar\_file

\end{fulllineitems}


\sphinxstepscope


\section{Settings module}
\label{\detokenize{settings:module-settings}}\label{\detokenize{settings:settings-module}}\label{\detokenize{settings::doc}}\index{module@\spxentry{module}!settings@\spxentry{settings}}\index{settings@\spxentry{settings}!module@\spxentry{module}}
\sphinxAtStartPar
This module handles the local settings for pymemdyn on your machine. 
The settings are mostly paths and run settings.


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{b}
\item\relax\sphinxstyleindexentry{broker}\sphinxstyleindexpageref{broker:\detokenize{module-broker}}
\item\relax\sphinxstyleindexentry{bw4posres}\sphinxstyleindexpageref{bw4posres:\detokenize{module-bw4posres}}
\indexspace
\bigletter{c}
\item\relax\sphinxstyleindexentry{complex}\sphinxstyleindexpageref{complex:\detokenize{module-complex}}
\indexspace
\bigletter{g}
\item\relax\sphinxstyleindexentry{groerrors}\sphinxstyleindexpageref{groerrors:\detokenize{module-groerrors}}
\item\relax\sphinxstyleindexentry{gromacs}\sphinxstyleindexpageref{gromacs:\detokenize{module-gromacs}}
\indexspace
\bigletter{m}
\item\relax\sphinxstyleindexentry{membrane}\sphinxstyleindexpageref{membrane:\detokenize{module-membrane}}
\indexspace
\bigletter{p}
\item\relax\sphinxstyleindexentry{protein}\sphinxstyleindexpageref{protein:\detokenize{module-protein}}
\indexspace
\bigletter{q}
\item\relax\sphinxstyleindexentry{queue}\sphinxstyleindexpageref{queue:\detokenize{module-queue}}
\indexspace
\bigletter{r}
\item\relax\sphinxstyleindexentry{recipes}\sphinxstyleindexpageref{recipes:\detokenize{module-recipes}}
\indexspace
\bigletter{s}
\item\relax\sphinxstyleindexentry{settings}\sphinxstyleindexpageref{settings:\detokenize{module-settings}}
\indexspace
\bigletter{u}
\item\relax\sphinxstyleindexentry{utils}\sphinxstyleindexpageref{utils:\detokenize{module-utils}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}