%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}


\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}

\global\renewcommand{\AA}{\text{\r{A}}}

\title{pymemdyn}
\date{Dec 11, 2023}
\release{2.0}
\author{H. Gutierrez de Teran \and X. Bello \and M. Esguerra \and R. L. van den Broek \and R.V. Küpper}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxstepscope


\chapter{PyMemDyn Version 2.0}
\label{\detokenize{installation:pymemdyn-version-2-0}}\label{\detokenize{installation::doc}}
\sphinxAtStartPar
PyMemDyn is a standalone \sphinxstyleemphasis{python} package to setup membrane molecular
dynamics calculations using the \sphinxstylestrong{GROMACS} set of programs. The package
can be used either in a desktop environment, or in a cluster with
popular queuing systems such as Torque/PBS or Slurm.

\sphinxAtStartPar
\sphinxstylestrong{PyMemDyn} is hosted in github at:

\sphinxAtStartPar
\sphinxurl{https://github.com/GPCR-ModSim/pymemdyn}

\sphinxAtStartPar
You can download any version of \sphinxstylestrong{PyMemDyn} by cloning the repository
to your local machine using git.

\sphinxAtStartPar
You will need to create a free personal account at github and send and
e\sphinxhyphen{}mail to: \sphinxhref{mailto:gpcruser@gmail.com}{gpcruser@gmail.com} requesting access to the code. After
request processing from us you will be given access to the free
repository.


\section{Dependencies}
\label{\detokenize{installation:dependencies}}
\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{GROMACS}
\item[] Pymemdyn is dependent on GROMACS. Download GROMACS
\sphinxhref{https://manual.gromacs.org/current/download.html}{here}.
Instructions for installation are
\sphinxhref{https://manual.gromacs.org/current/install-guide/index.html}{here}.
\end{DUlineblock}

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{LigParGen}
\item[] In order to automatically generate .itp files for ligands and
allosterics, the program ligpargen is used. Install using their
instructions: \sphinxurl{https://github.com/Isra3l/ligpargen}. Do not forget to
activate the conda environment in which you installed ligpargen
(\sphinxcode{\sphinxupquote{conda install py37}} if you followed the instructions) before
running pymemdyn. In case you are using a bash script, this should be
done inside the script. See also “ligpargen\_example” in the folder
examples.
\end{DUlineblock}

\sphinxAtStartPar
Testing was done using LigParGen v2.1 using BOSS5.0.

\sphinxAtStartPar
Pymemdyn can also be used without ligpargen installation, but then .itp
files containing the parameters for the ligand(s) should be provided.

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{MODELLER}
\item[] For replacing missing loops or missing side chains in the protein,
MODELLER is used. The download and installation guid for MODELLER
are available
\sphinxhref{https://salilab.org/modeller/download\_installation.html}{here}.
\end{DUlineblock}

\begin{DUlineblock}{0em}
\item[] \sphinxstylestrong{Queueing system}
\item[] A queuing system: although not strictly required, this is highly
advisable since an MD simulation of 2.5 nanoseconds will be
performed. However, if only membrane insertion and energy
minimization is requested, this requirement can be avoided.
Currently, the queuing systems supported include Slurm and PBS.
\end{DUlineblock}


\section{Installation}
\label{\detokenize{installation:installation}}
\sphinxAtStartPar
To install \sphinxstylestrong{PyMemDyn} follow these steps:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Clone \sphinxstylestrong{PyMemDyn} for python 3.7:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{git} \PYG{n}{clone} \PYG{n}{https}\PYG{p}{:}\PYG{o}{/}\PYG{o}{/}\PYG{n}{username}\PYG{n+nd}{@github}\PYG{o}{.}\PYG{n}{com}\PYG{o}{/}\PYG{n}{GPCR}\PYG{o}{\PYGZhy{}}\PYG{n}{ModSim}\PYG{o}{/}\PYG{n}{pymemdyn}\PYG{o}{.}\PYG{n}{git}
\end{sphinxVerbatim}

\sphinxAtStartPar
Make sure to change \sphinxstyleemphasis{username} to the one you have created at github.

\item {} 
\sphinxAtStartPar
The previous command will create a \sphinxstyleemphasis{pymemdyn} directory. Now you have
to tell your operating system how to find that folder. You achieve
this by declaring the location of the directory in a .bashrc file
.cshrc or .zshrc file in your home folder. An example of what you
will have to include in your .bashrc file follows:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
export PYMEMDYN=/home/username/software/pymemdyn
export PATH=\PYGZdl{}PYMEMDYN:\PYGZdl{}PATH
\end{sphinxVerbatim}

\sphinxAtStartPar
or if your shell is csh then in your .cshrc file you can add:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
setenv PYMEMDYN /home/username/software/pymemdyn
set path = (\PYGZdl{}path \PYGZdl{}PYMEMDYN)
\end{sphinxVerbatim}

\sphinxAtStartPar
Notice that I have cloned \sphinxstyleemphasis{pymemdyn} in the software folder in my
home folder, you will have to adapt this to wherever it is that you
downloaded your \sphinxstyleemphasis{pymemdyn} to.

\sphinxAtStartPar
After including the route to your \sphinxstyleemphasis{pymemdyn} directory in your
.bashrc file make sure to issue the command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{source} \PYG{o}{.}\PYG{n}{bashrc}
\end{sphinxVerbatim}

\sphinxAtStartPar
or open a new terminal.

\sphinxAtStartPar
To check if you have defined the route to the \sphinxstyleemphasis{pymemdyn} directory
correctly try to run the main program called pymemdyn in a terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pymemdyn} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{help}
\end{sphinxVerbatim}

\sphinxAtStartPar
You should obtain the following help output:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
usage: pymemdyn [\PYGZhy{}h] [\PYGZhy{}v] [\PYGZhy{}b OWN\PYGZus{}DIR] [\PYGZhy{}r REPO\PYGZus{}DIR] \PYGZhy{}p PDB [\PYGZhy{}l LIGAND]
             [\PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE] [\PYGZhy{}w WATERS] [\PYGZhy{}i IONS] [\PYGZhy{}\PYGZhy{}res RESTRAINT]
             [\PYGZhy{}f LOOP\PYGZus{}FILL] [\PYGZhy{}q QUEUE] [\PYGZhy{}d] [\PYGZhy{}\PYGZhy{}debugFast]

== Setup Molecular Dynamics for Membrane Proteins given a PDB. ==

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}version         show program\PYGZsq{}s version number and exit
  \PYGZhy{}b OWN\PYGZus{}DIR            Working dir if different from actual dir
  \PYGZhy{}r REPO\PYGZus{}DIR           Path to templates of fixed files. If not provided,
                        take the value from settings.TEMPLATES\PYGZus{}DIR.
  \PYGZhy{}p PDB                Name of the PDB file to insert into membrane for MD
                        (mandatory). Use the .pdb extension. (e.g. \PYGZhy{}p
                        myprot.pdb)
  \PYGZhy{}l LIGAND, \PYGZhy{}\PYGZhy{}lig LIGAND
                        Ligand identifiers of ligands present within the PDB
                        file. If multiple ligands are present, give a comma\PYGZhy{}
                        delimited list.
  \PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE    Charge of ligands for ligpargen (when itp file should
                        be generated). If multiple ligands are present, give a
                        comma\PYGZhy{}delimited list.
  \PYGZhy{}w WATERS, \PYGZhy{}\PYGZhy{}waters WATERS
                        Water identifiers of crystalized water molecules
                        present within the PDB file.
  \PYGZhy{}i IONS, \PYGZhy{}\PYGZhy{}ions IONS  Ion identifiers of crystalized ions present within the
                        PDB file.
  \PYGZhy{}\PYGZhy{}res RESTRAINT       Position restraints during MD production run. Options:
                        bw (Ballesteros\PYGZhy{}Weinstein Restrained Relaxation \PYGZhy{}
                        default), ca (C\PYGZhy{}Alpha Restrained Relaxation)
  \PYGZhy{}f LOOP\PYGZus{}FILL, \PYGZhy{}\PYGZhy{}loop\PYGZus{}fill LOOP\PYGZus{}FILL
                        Amount of Å per AA to fill cut loops. The total
                        distance is calculated from the coordinates of the
                        remaining residues. The AA contour length is 3.4\PYGZhy{}4.0
                        Å, To allow for flexibility in the loop, 2.0 Å/AA
                        (default) is suggested. (example: \PYGZhy{}f 2.0)
  \PYGZhy{}q QUEUE, \PYGZhy{}\PYGZhy{}queue QUEUE
                        Queueing system to use (slurm, pbs, pbs\PYGZus{}ib and svgd
                        supported)
  \PYGZhy{}d, \PYGZhy{}\PYGZhy{}debug
  \PYGZhy{}\PYGZhy{}debugFast           run pymemdyn in debug mode with less min and eq steps.
                        Do not use for simulation results!
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
Updates are very easy thanks to the git versioning system. Once
\sphinxstylestrong{PyMemDyn} has been downloaded (cloned) into its own \sphinxstyleemphasis{pymemdyn}
folder you just have to move to it and pull the newest changes:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{cd} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{username}\PYG{o}{/}\PYG{n}{software}\PYG{o}{/}\PYG{n}{pymemdyn}
\PYG{n}{git} \PYG{n}{pull}
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
You can also clone older stable versions of \sphinxstylestrong{PyMemDyn}. For example
the stable version 1.4 which works well and has been tested
extensively again GROMACS version 4.6.7 can be cloned with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{git} \PYG{n}{clone} \PYG{n}{https}\PYG{p}{:}\PYG{o}{/}\PYG{o}{/}\PYG{n}{username}\PYG{n+nd}{@github}\PYG{o}{.}\PYG{n}{com}\PYG{o}{/}\PYG{n}{GPCR}\PYG{o}{\PYGZhy{}}\PYG{n}{ModSim}\PYG{o}{/}\PYG{n}{pymemdyn}\PYG{o}{.}\PYG{n}{git} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{branch} \PYG{n}{stable}\PYG{o}{/}\PYG{l+m+mf}{1.4} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{single}\PYG{o}{\PYGZhy{}}\PYG{n}{branch} \PYG{n}{pymemdyn}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.4}
\end{sphinxVerbatim}

\sphinxAtStartPar
Now you will have to change your .bashrc or .cshrc files in your home
folder accordingly.

\item {} 
\sphinxAtStartPar
To make sure that your GROMACS installation is understood by
\sphinxstylestrong{PyMemDyn} you will need to specify the path to where GROMACS is
installed in your system. To do this you will need to edit the
settings.py file with any text editor (vi and emacs are common
options in the unix environment). Make sure that only one line is
uncommented, looking like: GROMACS\_PATH = /opt/gromacs\sphinxhyphen{}2021/bin
Provided that in your case gromacs is installed in /opt. The program
will prepend this line to the binaries names, so calling
/opt/gromacs\sphinxhyphen{}2021/bin/gmx should point to that binary.

\end{enumerate}


\section{Modules}
\label{\detokenize{installation:modules}}

\subsection{Modeling Modules}
\label{\detokenize{installation:modeling-modules}}
\sphinxAtStartPar
The following modules define the objects to be modeled.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{protein.py}. This module defines the ProteinComplex, Protein,
Monomer, Dimer, Compound, Ligand, CrystalWaters, Ions, Cholesterol,
Lipids, and Alosteric objects. These objects are started with the
required files, and can then be passed to other objects.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{membrane.py}. Defines the cellular membrane.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{complex.py}. Defines the full complex, protein + membrane.
It can include any of the previous objects.

\end{itemize}


\subsection{Auxiliary Modules}
\label{\detokenize{installation:auxiliary-modules}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{checks.py}. Checks continuity of the protein and composition of the
residues.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{aminoAcids.py}. Contains the amino acids class that defines the 1\sphinxhyphen{}letter
and three letter codes, along with the number of different atoms per residue.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{queue.py}.   Queue  manager.  That  is,  it  receives  objects to  be
executed.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{recipes.py}.   Applies  step by  step instructions  for  carrying a
modeling  step.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{bw4posres.py}. Creates a set of distance restraints based on
Ballesteros\sphinxhyphen{}Weinstein identities which are gathered by alignment to a
multiple\sphinxhyphen{}sequence alignment using clustalw.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{utils.py}.  Puts the  functions done by the previous objects on demand.
For example, manipulate files, copy  folders, call functions or classes from
standalone modules like bw4posres.py, etc.

\item {} 
\sphinxAtStartPar
\sphinxstylestrong{settings.py} This modules sets up the main environment variables needed
to run the calculation, for example, the path to the gromacs binaries.

\end{itemize}


\subsection{Execution Modules}
\label{\detokenize{installation:execution-modules}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{gromacs.py}. Defines the Gromacs and Wrapper objects. * Gromacs
will load the objects to be modeled, the modeling recipe, and run it.
* Wrapper is a proxy for gromacs commands. When a recipe entry is
sent to it this returns the command to be run.

\end{itemize}


\subsection{Executable}
\label{\detokenize{installation:executable}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstylestrong{pymemdyn} The main program to call which sends the run to a
cluster.

\end{itemize}

\sphinxAtStartPar
More information about all modules can be found in the Modules chapter.

\sphinxstepscope


\chapter{Manual}
\label{\detokenize{manual:manual}}\label{\detokenize{manual::doc}}
\sphinxAtStartPar
The fully automated pipeline available by using \sphinxstylestrong{PyMemDyn} allows any
researcher, without prior experience in computational chemistry, to
perform an otherwise tedious and complex process of membrane insertion
and thorough MD equilibration, as outlined in Figure 1.

\noindent\sphinxincludegraphics{{pymemdyn_scheme}.png}

\sphinxAtStartPar
In the simplest scenario, only the receptor structure is considered. In
such case the GPCR is automatically surrounded by a pre\sphinxhyphen{}equilibrated
POPC (Palmitoyl\sphinxhyphen{}Oleoyl\sphinxhyphen{}Phosphatidyl\sphinxhyphen{}Choline) membrane model in a way
that the TM (Trans\sphinxhyphen{}Membrane) bundle is parallel to the vertical axis of
the membrane. The system is then soaked with bulk water and inserted
into an hexagonal prism\sphinxhyphen{}shaped box, which is energy\sphinxhyphen{}minimized and
carefully equilibrated in the framework of periodic boundary conditions
(PBC). A thorough MD equilibration protocol lasting 2.5 ns follows.

\sphinxAtStartPar
But the simulation of an isolated receptor can only account for one part
of the problem, and the influence of different non\sphinxhyphen{}protein elements in
receptor dynamics such as the orthosteric (primary) ligand, allosteric
modulator, or even specific cholesterol, lipid, water or ion molecules
are key for a more comprehensive characterization of GPCRs. \sphinxstylestrong{PyMemDyn}
can explicitly handle these elements allowing a broader audience in the
field of GPCRs to use molecular dynamics simulations. These molecules
should be uploaded in the same PDB file of the receptor, so they are
properly integrated into the membrane insertion protocol described above.
Force\sphinxhyphen{}field associated files are generated automatically with LigParGen.
In addition, it is also possible to perform MD simulations of other
membrane proteins, provided that a proper dimerization model exists (i.e.,
coming from X\sphinxhyphen{}ray crystallography or from a protein\sphinxhyphen{}protein docking
protocol). The ease of use, flexibility and public availability of the
\sphinxstylestrong{PyMemDyn} library makes it a unique tool for researchers in the
receptor field interested in exploring dynamic processes of these receptors.


\section{Getting started}
\label{\detokenize{manual:getting-started}}
\sphinxAtStartPar
After you have finished the installation of \sphinxstylestrong{PyMemDyn} you can check
if the installation has succeeded with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pymemdyn} \PYG{o}{\PYGZhy{}}\PYG{n}{h}
\end{sphinxVerbatim}

\sphinxAtStartPar
This should print the following message:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
usage: pymemdyn [\PYGZhy{}h] [\PYGZhy{}v] [\PYGZhy{}b OWN\PYGZus{}DIR] [\PYGZhy{}r REPO\PYGZus{}DIR] \PYGZhy{}p PDB [\PYGZhy{}l LIGAND]
                [\PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE] [\PYGZhy{}w WATERS] [\PYGZhy{}i IONS] [\PYGZhy{}\PYGZhy{}res RESTRAINT]
                [\PYGZhy{}f LOOP\PYGZus{}FILL] [\PYGZhy{}q QUEUE] [\PYGZhy{}d] [\PYGZhy{}\PYGZhy{}debugFast]

== Setup Molecular Dynamics for Membrane Proteins given a PDB. ==

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}version         show program\PYGZsq{}s version number and exit
  \PYGZhy{}b OWN\PYGZus{}DIR            Working dir if different from actual dir
  \PYGZhy{}r REPO\PYGZus{}DIR           Path to templates of fixed files. If not provided,
                        take the value from settings.TEMPLATES\PYGZus{}DIR.
  \PYGZhy{}p PDB                Name of the PDB file to insert into membrane for MD
                        (mandatory). Use the .pdb extension. (e.g. \PYGZhy{}p
                        myprot.pdb)
  \PYGZhy{}l LIGAND, \PYGZhy{}\PYGZhy{}lig LIGAND
                        Ligand identifiers of ligands present within the PDB
                        file. If multiple ligands are present, give a comma\PYGZhy{}
                        delimited list.
  \PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE    Charge of ligands for ligpargen (when itp file should
                        be generated). If multiple ligands are present, give a
                        comma\PYGZhy{}delimited list.
  \PYGZhy{}w WATERS, \PYGZhy{}\PYGZhy{}waters WATERS
                        Water identifiers of crystalized water molecules
                        present within the PDB file.
  \PYGZhy{}i IONS, \PYGZhy{}\PYGZhy{}ions IONS  Ion identifiers of crystalized ions present within the
                        PDB file.
  \PYGZhy{}\PYGZhy{}res RESTRAINT       Position restraints during MD production run. Options:
                        bw (Ballesteros\PYGZhy{}Weinstein Restrained Relaxation \PYGZhy{}
                        default), ca (C\PYGZhy{}Alpha Restrained Relaxation)
  \PYGZhy{}f LOOP\PYGZus{}FILL, \PYGZhy{}\PYGZhy{}loop\PYGZus{}fill LOOP\PYGZus{}FILL
                        Amount of Å per AA to fill cut loops. The total
                        distance is calculated from the coordinates of the
                        remaining residues. The AA contour length is 3.4\PYGZhy{}4.0
                        Å, To allow for flexibility in the loop, 2.0 Å/AA
                        (default) is suggested. (example: \PYGZhy{}f 2.0)
  \PYGZhy{}q QUEUE, \PYGZhy{}\PYGZhy{}queue QUEUE
                        Queueing system to use (slurm, pbs, pbs\PYGZus{}ib and svgd
                        supported)
  \PYGZhy{}d, \PYGZhy{}\PYGZhy{}debug
  \PYGZhy{}\PYGZhy{}debugFast           run pymemdyn in debug mode with less min and eq steps.
                        Do not use for simulation results!
\end{sphinxVerbatim}

\sphinxAtStartPar
Don’t forget to activate your conda environment if needed.

\sphinxAtStartPar
Now let’s say we want to run a system that consists of a membrane protein, two
ligands, an ion and some crystallized waters. We need to know a couple of
things about the system: the three\sphinxhyphen{}letter abbreviations of the ligands, ion
and waters, the charge of the ligands. So let’s say that in our file we have
ligand 1 “LIG”, ligand 2 “LIH”, ion “NA” and water “POH” and ligand 2 has a
charge of \sphinxhyphen{}1. Then we can run pymemdyn with the following line:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pymemdyn} \PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{file}\PYG{o}{.}\PYG{n}{pdb} \PYG{o}{\PYGZhy{}}\PYG{n}{l} \PYG{n}{LIG}\PYG{p}{,}\PYG{n}{LIH} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{lc} \PYG{l+m+mi}{0}\PYG{p}{,}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1} \PYG{o}{\PYGZhy{}}\PYG{n}{i} \PYG{n}{NA} \PYG{o}{\PYGZhy{}}\PYG{n}{w} \PYG{n}{POH} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{res} \PYG{n}{ca}
\end{sphinxVerbatim}

\sphinxAtStartPar
In case our protein has a missing loop that needs to be filled with Modeller,
we can also define the number of Å per AA with \sphinxhyphen{}f.


\section{Running with queues}
\label{\detokenize{manual:running-with-queues}}
\sphinxAtStartPar
About 90\% of the time you will want to use some queueing system. We
deal with queue systems tweaks as we stumble into them and it’s out of
our scope to cover them all. If you take a look at the source code dir,
you’ll found some files called “run\_pbs.sh”, “run\_svgd.sh” and so on.
Also there are specific queue objects in the source file queue.py we
have to tweak for every and each queue. In you want to run your
simulation in a supported queue, copy the “run\_queuename.sh” file to
your working directory, and edit it. E.g. the workdir to run an A2a.pdb
simulation in svgd.cesga.es looks like: . .. A2a.pdb run\_svgd.sh And
run\_svgd.sh looks like:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZbs{}\PYGZdl{}!/bin/bash
module   load  python/3.7
module   load  gromacs/2021
python \PYGZti{}/bin/pymoldyn/pymemdyn  \PYGZhy{}p  a2a.pdb
\end{sphinxVerbatim}

\sphinxAtStartPar
Now we just launch this script with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{qsub} \PYG{o}{\PYGZhy{}}\PYG{n}{l} \PYG{n}{arch}\PYG{o}{=}\PYG{n}{amd}\PYG{p}{,}\PYG{n}{num\PYGZus{}proc}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{p}{,}\PYG{n}{s\PYGZus{}rt}\PYG{o}{=}\PYG{l+m+mi}{50}\PYG{p}{:}\PYG{l+m+mi}{00}\PYG{p}{:}\PYG{l+m+mi}{00}\PYG{p}{,}\PYG{n}{s\PYGZus{}vmem}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{n}{G}\PYG{p}{,}\PYG{n}{h\PYGZus{}fsize}\PYG{o}{=}\PYG{l+m+mi}{1}\PYG{n}{G} \PYG{o}{\PYGZhy{}}\PYG{n}{pe} \PYG{n}{mpi} \PYG{l+m+mi}{8} \PYG{n}{run\PYGZus{}svgd}\PYG{o}{.}\PYG{n}{sh}
\end{sphinxVerbatim}

\sphinxAtStartPar
and wait for the results. Note that we launch 1 process, but flag the
run as mpi with reservation of 8 cores in SVGD queue.


\section{Debugging}
\label{\detokenize{manual:debugging}}
\sphinxAtStartPar
To also log ‘debug’\sphinxhyphen{}messages to the log file (log.log), activate the
debug mode with \textendash{}debug:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pymemdyn} \PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{proteinComplex}\PYG{o}{.}\PYG{n}{pdb} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{debug}
\end{sphinxVerbatim}

\sphinxAtStartPar
If you are to set up a new system, it is a good idea to just run a few
steps of each stage in the equilibration protocol just to test that the
pdb file is read correctly and the membrane\sphinxhyphen{}insertion protocol works
fine and the system can be minimized and does not “explode” during the
equilibration protocol (i.e., detect if atom clashes and so on exist on
your system).

\sphinxAtStartPar
To do this, use the \textendash{}debugFast option, like:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{pymemdyn} \PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{gpcr}\PYG{o}{.}\PYG{n}{pdb} \PYG{o}{\PYGZhy{}}\PYG{n}{l} \PYG{n}{LIG} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{waters} \PYG{n}{HOH} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{debugFast}
\end{sphinxVerbatim}

\sphinxAtStartPar
If everythings works fine, you will see the list of output directories
and files just as in a regular equilibration protocol, but with much
smaller files (since we only use here 1000 steps of MD in each stage).
NOTE that sometimes, due to the need of a smooth equilibration procedure
(i.e. when a new ligand is introduced in the binding site without
further refinement of the complex, or with slight clashes of existing
water molecules) this kind of debugging equilibration procedure might
crash during the first stages due to hot atoms or LINCS failure. This is
normal, and you have two options: i) trust that the full equilibration
procedure will fix the steric clashes in your starting system, and then
directly run the pymemdyn without the debugging option, or ii) identify
the hot atoms (check the mdrun.log file in the last subdirectory that
was written in your output (generally eq/mdrun.log and look for “LINCS
WARNING”). What if you want to check partial functions of pymoldyn? In
order to do this you must edit the file pymemdyn and change:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Line 260 comment with “\#” this line {[}that states: run.clean(){]}, which
is the one that deletes all the output files present in the working
directory.

\item {} 
\sphinxAtStartPar
In the last two lines of this file, comment (add a “\#”) the line:
run.moldyn()

\item {} 
\sphinxAtStartPar
And uncomment (remove the “\#”) the line: run.light\_moldyn()

\item {} 
\sphinxAtStartPar
In the line 153 and within that block (ligh\_moldyn) change the lines
stating steps = {[}“xxxx”{]} and include only those steps that you want
to test, which should be within a list of strings.

\end{enumerate}

\sphinxAtStartPar
For the sake of clarity, these have been subdivided in two lines:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{line} \PYG{l+m+mi}{1}\PYG{o}{\PYGZhy{}}  \PYG{n}{steps} \PYG{o}{=} \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Init}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Minimization}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Equilibration}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Relax}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CARelax}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
Here you remove those strings that you do not want to be executed, i.e.
if only membrane insertion and minimization is wished, remove
“Equilibration”, “Relax”, “CARelax” so the line states:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{steps}   \PYG{o}{=}   \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Init}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,}  \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Minimization}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}

\PYG{n}{line}  \PYG{l+m+mi}{2}\PYG{o}{\PYGZhy{}}   \PYG{n}{steps}  \PYG{o}{=}  \PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{CollectResults}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}
\end{sphinxVerbatim}

\sphinxAtStartPar
This only accounts for the preparation of the output files for analysis,
so if you only are interested on this stage, comment the previous line.
The last assignment is the one that runs. NOTE that you must know what
you do, otherwise you might have crashes in the code if needed files to
run intermediate stages are missing!


\section{Output}
\label{\detokenize{manual:output}}
\sphinxAtStartPar
The performed equilibration includes the following stages:


\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\centering
\sphinxcapstartof{table}
\sphinxthecaptionisattop
\sphinxcaption{Output}\label{\detokenize{manual:id1}}
\sphinxaftertopcaption
\begin{tabular}[t]{|\X{20}{100}|\X{35}{100}|\X{25}{100}|\X{20}{100}|}
\sphinxtoprule
\sphinxstyletheadfamily 
\sphinxAtStartPar
STAGE
&\sphinxstyletheadfamily 
\sphinxAtStartPar
RESTRAINED ATOMS
&\sphinxstyletheadfamily 
\sphinxAtStartPar
FORCE CONSTANT
&\sphinxstyletheadfamily 
\sphinxAtStartPar
TIME
\\
\sphinxmidrule
\sphinxtableatstartofbodyhook&&
\sphinxAtStartPar
kJ/(mol A ·nm\textasciicircum{}2)
&
\sphinxAtStartPar
ns
\\
\sphinxhline
\sphinxAtStartPar
Minimization
&&&
\sphinxAtStartPar
(Max. 500 steps)
\\
\sphinxhline
\sphinxAtStartPar
Equil. 1
&
\sphinxAtStartPar
Protein Heavy Atoms
&
\sphinxAtStartPar
1000
&
\sphinxAtStartPar
0.5
\\
\sphinxhline
\sphinxAtStartPar
Equil. 2
&
\sphinxAtStartPar
Protein Heavy Atoms
&
\sphinxAtStartPar
800
&
\sphinxAtStartPar
0.5
\\
\sphinxhline
\sphinxAtStartPar
Equil. 3
&
\sphinxAtStartPar
Protein Heavy Atoms
&
\sphinxAtStartPar
600
&
\sphinxAtStartPar
0.5
\\
\sphinxhline
\sphinxAtStartPar
Equil. 4
&
\sphinxAtStartPar
Protein Heavy Atoms
&
\sphinxAtStartPar
400
&
\sphinxAtStartPar
0.5
\\
\sphinxhline
\sphinxAtStartPar
Equil. 5
&
\sphinxAtStartPar
Protein Heavy Atoms
&
\sphinxAtStartPar
200
&
\sphinxAtStartPar
0.5
\\
\sphinxhline
\sphinxAtStartPar
Equil. 6
&
\sphinxAtStartPar
Venkatakrishnan Pairs /
&
\sphinxAtStartPar
200 /
&
\sphinxAtStartPar
2.5 /
\\
\sphinxhline
\sphinxAtStartPar
Equil. 6
&
\sphinxAtStartPar
C\sphinxhyphen{}alpha Atoms
&
\sphinxAtStartPar
200
&
\sphinxAtStartPar
2.5
\\
\sphinxbottomrule
\end{tabular}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
In this folder you will find several files related to this simulation:


\subsection{INPUT:}
\label{\detokenize{manual:input}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZhy{}} \PYG{n}{popc}\PYG{o}{.}\PYG{n}{itp}              \PYG{c+c1}{\PYGZsh{} Topology of the lipids}
\PYG{o}{\PYGZhy{}} \PYG{n}{ffoplsaa\PYGZus{}mod}\PYG{o}{.}\PYG{n}{itp}      \PYG{c+c1}{\PYGZsh{} Modified OPLSAA\PYGZhy{}FF, to account for lipid modifications}
\PYG{o}{\PYGZhy{}} \PYG{n}{ffoplsaabon\PYGZus{}mod}\PYG{o}{.}\PYG{n}{itp}   \PYG{c+c1}{\PYGZsh{} Modified OPLSAA\PYGZhy{}FF(bonded), to account for lipid modifications}
\PYG{o}{\PYGZhy{}} \PYG{n}{ffoplsaanb\PYGZus{}mod}\PYG{o}{.}\PYG{n}{itp}    \PYG{c+c1}{\PYGZsh{} Modified OPLSAA\PYGZhy{}FF(non\PYGZhy{}bonded), to account for lipid modifications}
\PYG{o}{\PYGZhy{}} \PYG{n}{topol}\PYG{o}{.}\PYG{n}{tpr}             \PYG{c+c1}{\PYGZsh{} Input for the first equilibration stage}
\PYG{o}{\PYGZhy{}} \PYG{n}{topol}\PYG{o}{.}\PYG{n}{top}             \PYG{c+c1}{\PYGZsh{} Topology of the system}
\PYG{o}{\PYGZhy{}} \PYG{n}{protein}\PYG{o}{.}\PYG{n}{itp}           \PYG{c+c1}{\PYGZsh{} Topology of the protein}
\PYG{o}{\PYGZhy{}} \PYG{n}{index}\PYG{o}{.}\PYG{n}{ndx}             \PYG{c+c1}{\PYGZsh{} Index file with appropriate groups for GROMACS}
\PYG{o}{\PYGZhy{}} \PYG{n}{prod}\PYG{o}{.}\PYG{n}{mdp}              \PYG{c+c1}{\PYGZsh{} Example of a parameter file to configure a production run (see TIPS)}
\end{sphinxVerbatim}


\subsection{STRUCTURES:}
\label{\detokenize{manual:structures}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZhy{}} \PYG{n}{hexagon}\PYG{o}{.}\PYG{n}{pdb}           \PYG{c+c1}{\PYGZsh{} Initial structure of the system, with the receptor centered in the box}
\PYG{o}{\PYGZhy{}} \PYG{n}{confout}\PYG{o}{.}\PYG{n}{gro}           \PYG{c+c1}{\PYGZsh{} Final structure of the system (see TIPS)}
\PYG{o}{\PYGZhy{}} \PYG{n}{load\PYGZus{}gpcr}\PYG{o}{.}\PYG{n}{pml}         \PYG{c+c1}{\PYGZsh{} Loads the initial structure and the trajectory in pymol}
\end{sphinxVerbatim}


\subsection{TRAJECTORY FILES:}
\label{\detokenize{manual:trajectory-files}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{o}{\PYGZhy{}} \PYG{n}{traj\PYGZus{}pymol}\PYG{o}{.}\PYG{n}{xtc}        \PYG{c+c1}{\PYGZsh{} Trajectory of the whole system for visualization in pymol. 1 snapshot/100 ps}
\PYG{o}{\PYGZhy{}} \PYG{n}{traj\PYGZus{}EQ}\PYG{o}{.}\PYG{n}{xtc}           \PYG{c+c1}{\PYGZsh{} Trajectory of the whole system in .xtc format: 1 snapshot/50 ps}
\PYG{o}{\PYGZhy{}} \PYG{n}{ener\PYGZus{}EQ}\PYG{o}{.}\PYG{n}{edr}           \PYG{c+c1}{\PYGZsh{} Energy file of the trajectory}
\PYG{o}{\PYGZhy{}} \PYG{n}{load\PYGZus{}gpcr}\PYG{o}{.}\PYG{n}{pml}         \PYG{c+c1}{\PYGZsh{} Script to load the equilibration trajectory in pymol.}
\end{sphinxVerbatim}


\subsection{REPORTS:}
\label{\detokenize{manual:reports}}
\begin{DUlineblock}{0em}
\item[] In the “reports” subfolder, you will find the following files:
\end{DUlineblock}


\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\centering
\begin{tabular}[t]{|\X{45}{100}|\X{55}{100}|}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
tot\_ener.xvg, tot\_ener.log
&
\sphinxAtStartPar
System total energy plot and log
\\
\sphinxhline
\sphinxAtStartPar
temp.xvg, temp.log
&
\sphinxAtStartPar
System temperature plot and log
\\
\sphinxhline
\sphinxAtStartPar
pressure.xvg, pressure.log
&
\sphinxAtStartPar
System pressure plot and log
\\
\sphinxhline
\sphinxAtStartPar
volume.xvg, volume.log
&
\sphinxAtStartPar
System volume plot and log
\\
\sphinxhline
\sphinxAtStartPar
rmsd\sphinxhyphen{}all\sphinxhyphen{}atom\sphinxhyphen{}vs\sphinxhyphen{}start
&
\sphinxAtStartPar
All atoms RMSD plot
\\
\sphinxhline
\sphinxAtStartPar
rmsd\sphinxhyphen{}backbone\sphinxhyphen{}vs\sphinxhyphen{}start.xvg
&
\sphinxAtStartPar
Backbone RMSD plot
\\
\sphinxhline
\sphinxAtStartPar
rmsd\sphinxhyphen{}calpha\sphinxhyphen{}vs\sphinxhyphen{}start.xvg
&
\sphinxAtStartPar
C\sphinxhyphen{}Alpha RMSD plot
\\
\sphinxhline
\sphinxAtStartPar
rmsf\sphinxhyphen{}per\sphinxhyphen{}residue.xvg
&
\sphinxAtStartPar
Residue RMSF plot
\\
\sphinxbottomrule
\end{tabular}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}


\subsection{LOGS:}
\label{\detokenize{manual:logs}}
\sphinxAtStartPar
The logger in pymemdyn will write log messages to the file log.log, which
is regenerated every run.

\begin{DUlineblock}{0em}
\item[] In the “logs” subfolder, you will find the log files of mdrun:
\end{DUlineblock}


\begin{savenotes}\sphinxattablestart
\sphinxthistablewithglobalstyle
\centering
\begin{tabular}[t]{|\X{45}{100}|\X{55}{100}|}
\sphinxtoprule
\sphinxtableatstartofbodyhook
\sphinxAtStartPar
eq\_\{force\_constant\}.log
&
\sphinxAtStartPar
log of stages with restrained heavy atoms of the receptor
\\
\sphinxhline
\sphinxAtStartPar
eqCA.log
&
\sphinxAtStartPar
log of the stage with restrained C\sphinxhyphen{}alfa atoms of the receptor
\\
\sphinxbottomrule
\end{tabular}
\sphinxtableafterendhook\par
\sphinxattableend\end{savenotes}

\sphinxAtStartPar
\sphinxstylestrong{NOTE ON GROMACS METHODS} To integrate the equations of motion we have
selected the leap\sphinxhyphen{}frog integrator with a 2 femtosecond timestep.
Longe\sphinxhyphen{}range electrostatic interactions in periodic boundary conditions
are treated with the particle mesh Ewald method. We use a Nose\sphinxhyphen{}Hoover
thermostat with a tau\_t of 0.5 picoseconds and a Parinello\sphinxhyphen{}Rahman
barostat with a tau\_p of 2.0. The pressure coupling is semiisotropic,
meaning that it’s isotropic in the x and y directions but different in
the z direction. Since we are using pressure coupling we are working
with an NPT ensemble. This is done both in the all\sphinxhyphen{}atom restrained steps
and in the alpha\sphinxhyphen{}carbon atom restrained part. All of these details are
more explicitly stated in the Rodriguez et al. {[}1{]} publication.

\sphinxAtStartPar
\sphinxstylestrong{TIPS}

\sphinxAtStartPar
NOTE: these tips work for GROMACS version \textgreater{}= 4.5 and \textless{} 5.0. For later
versions, adjustments are required, but the principle remains the same.
\begin{itemize}
\item {} 
\sphinxAtStartPar
If you want to configure a .tpr input file for a \sphinxstylestrong{production} run,
you can use the template ‘prod.mdp’ file by introducing the number of
steps (nsteps), and thus the simulation time, you want to run.

\end{itemize}

\sphinxAtStartPar
After that, you just have to type:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{grompp} \PYG{o}{\PYGZhy{}}\PYG{n}{f} \PYG{n}{prod}\PYG{o}{.}\PYG{n}{mdp} \PYG{o}{\PYGZhy{}}\PYG{n}{c} \PYG{n}{confout}\PYG{o}{.}\PYG{n}{gro} \PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{topol}\PYG{o}{.}\PYG{n}{top} \PYG{o}{\PYGZhy{}}\PYG{n}{n} \PYG{n}{index}\PYG{o}{.}\PYG{n}{ndx} \PYG{o}{\PYGZhy{}}\PYG{n}{o} \PYG{n}{topol\PYGZus{}prod}\PYG{o}{.}\PYG{n}{tpr}
\PYG{n}{mdrun} \PYG{o}{\PYGZhy{}}\PYG{n}{s} \PYG{n}{topol\PYGZus{}prod}\PYG{o}{.}\PYG{n}{tpr} \PYG{o}{\PYGZhy{}}\PYG{n}{o} \PYG{n}{traj}\PYG{o}{.}\PYG{n}{trr} \PYG{o}{\PYGZhy{}}\PYG{n}{e} \PYG{n}{ener}\PYG{o}{.}\PYG{n}{edr} \PYG{o}{\PYGZhy{}}\PYG{n}{c} \PYG{n}{confout}\PYG{o}{.}\PYG{n}{gro} \PYG{o}{\PYGZhy{}}\PYG{n}{g} \PYG{n}{production}\PYG{o}{.}\PYG{n}{log} \PYG{o}{\PYGZhy{}}\PYG{n}{x} \PYG{n}{traj\PYGZus{}prod}\PYG{o}{.}\PYG{n}{xtc}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
If you want to create a PDB file of your system after the
equilibration, with the receptor centered in the box, type:

\sphinxAtStartPar
echo 1 0 | trjconv \sphinxhyphen{}pbc mol \sphinxhyphen{}center \sphinxhyphen{}ur compact \sphinxhyphen{}f confout.gro \sphinxhyphen{}o
confout.pdb

\item {} 
\sphinxAtStartPar
If you want to create an xmgrace graph of the root mean square
deviation for c\sphinxhyphen{}alpha atoms in the 5.0 ns of simulation you can use:

\sphinxAtStartPar
echo 3 3 | g\_rms \sphinxhyphen{}f traj\_EQ.xtc \sphinxhyphen{}s topol.tpr \sphinxhyphen{}o
rmsd\sphinxhyphen{}calpha\sphinxhyphen{}vs\sphinxhyphen{}start.xvg

\item {} 
\sphinxAtStartPar
You may want to get a pdb file of your last frame. You can first
check the total time of your trajectory and then use this time to
request the last frame with:

\sphinxAtStartPar
gmxcheck \sphinxhyphen{}f traj\_pymol.xtc echo 1 | trjconv \sphinxhyphen{}b 5000 \sphinxhyphen{}e 5000 \sphinxhyphen{}f
traj\_pymol.xtc \sphinxhyphen{}o last51.pdb

\end{itemize}


\section{References}
\label{\detokenize{manual:references}}
\begin{DUlineblock}{0em}
\item[] {[}1{]} Rodríguez D., Piñeiro A. and Gutiérrez\sphinxhyphen{}de\sphinxhyphen{}Terán H.
\item[] Molecular Dynamics Simulations Reveal Insights into Key Structural
Elements of Adenosine Receptors
\item[] Biochemistry (2011), 50, 4194\sphinxhyphen{}208.
\end{DUlineblock}

\sphinxstepscope


\chapter{Modules}
\label{\detokenize{modules:modules}}\label{\detokenize{modules::doc}}
\sphinxstepscope


\section{Pymemdyn}
\label{\detokenize{pymemdyn:pymemdyn}}\label{\detokenize{pymemdyn::doc}}
\sphinxAtStartPar
This is the main script for the pymemdyn commandline tool.
In this script the following things are accomplished:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Command line arguments are parsed.

\item {} 
\sphinxAtStartPar
(If necessary) a working directory is created.

\item {} 
\sphinxAtStartPar
(If necessary) previous Run files are removed.

\item {} 
\sphinxAtStartPar
A run is done.

\end{enumerate}


\subsection{Usage}
\label{\detokenize{pymemdyn:usage}}
\begin{sphinxVerbatim}[commandchars=\\\{\}]
usage: pymemdyn [\PYGZhy{}h] [\PYGZhy{}v] [\PYGZhy{}b OWN\PYGZus{}DIR] [\PYGZhy{}r REPO\PYGZus{}DIR] \PYGZhy{}p PDB [\PYGZhy{}l LIGAND]
             [\PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE] [\PYGZhy{}w WATERS] [\PYGZhy{}i IONS] [\PYGZhy{}\PYGZhy{}res RESTRAINT]
             [\PYGZhy{}f LOOP\PYGZus{}FILL] [\PYGZhy{}q QUEUE] [\PYGZhy{}d] [\PYGZhy{}\PYGZhy{}debugFast]

== Setup Molecular Dynamics for Membrane Proteins given a PDB. ==

optional arguments:
  \PYGZhy{}h, \PYGZhy{}\PYGZhy{}help            show this help message and exit
  \PYGZhy{}v, \PYGZhy{}\PYGZhy{}version         show program\PYGZsq{}s version number and exit
  \PYGZhy{}b OWN\PYGZus{}DIR            Working dir if different from actual dir
  \PYGZhy{}r REPO\PYGZus{}DIR           Path to templates of fixed files. If not provided,
                        take the value from settings.TEMPLATES\PYGZus{}DIR.
  \PYGZhy{}p PDB                Name of the PDB file to insert into membrane for MD
                        (mandatory). Use the .pdb extension. (e.g. \PYGZhy{}p
                        myprot.pdb)
  \PYGZhy{}l LIGAND, \PYGZhy{}\PYGZhy{}lig LIGAND
                        Ligand identifiers of ligands present within the PDB
                        file. If multiple ligands are present, give a comma\PYGZhy{}
                        delimited list.
  \PYGZhy{}\PYGZhy{}lc LIGAND\PYGZus{}CHARGE    Charge of ligands for ligpargen (when itp file should
                        be generated). If multiple ligands are present, give a
                        comma\PYGZhy{}delimited list.
  \PYGZhy{}w WATERS, \PYGZhy{}\PYGZhy{}waters WATERS
                        Water identifiers of crystalized water molecules
                        present within the PDB file.
  \PYGZhy{}i IONS, \PYGZhy{}\PYGZhy{}ions IONS  Ion identifiers of crystalized ions present within the
                        PDB file.
  \PYGZhy{}\PYGZhy{}res RESTRAINT       Position restraints during MD production run. Options:
                        bw (Ballesteros\PYGZhy{}Weinstein Restrained Relaxation \PYGZhy{}
                        default), ca (C\PYGZhy{}Alpha Restrained Relaxation)
  \PYGZhy{}f LOOP\PYGZus{}FILL, \PYGZhy{}\PYGZhy{}loop\PYGZus{}fill LOOP\PYGZus{}FILL
                        Amount of Å per AA to fill cut loops. The total
                        distance is calculated from the coordinates of the
                        remaining residues. The AA contour length is 3.4\PYGZhy{}4.0
                        Å, To allow for flexibility in the loop, 2.0 Å/AA
                        (default) is suggested. (example: \PYGZhy{}f 2.0)
  \PYGZhy{}q QUEUE, \PYGZhy{}\PYGZhy{}queue QUEUE
                        Queueing system to use (slurm, pbs, pbs\PYGZus{}ib and svgd
                        supported)
  \PYGZhy{}d, \PYGZhy{}\PYGZhy{}debug
  \PYGZhy{}\PYGZhy{}debugFast           run pymemdyn in debug mode with less min and eq steps.
                        Do not use for simulation results!
\end{sphinxVerbatim}

\sphinxstepscope


\section{Run module}
\label{\detokenize{run:module-run}}\label{\detokenize{run:run-module}}\label{\detokenize{run::doc}}\index{module@\spxentry{module}!run@\spxentry{run}}\index{run@\spxentry{run}!module@\spxentry{module}}\index{Run (class in run)@\spxentry{Run}\spxextra{class in run}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{run:run.Run}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{run.}}\sphinxbfcode{\sphinxupquote{Run}}}{\emph{\DUrole{n}{pdb}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{clean() (run.Run method)@\spxentry{clean()}\spxextra{run.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{run:run.Run.clean}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{clean}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Removes all previously generated files

\end{fulllineitems}

\index{moldyn() (run.Run method)@\spxentry{moldyn()}\spxextra{run.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{run:run.Run.moldyn}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{moldyn}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Run all steps in a molecular dynamics simulation of a membrane protein

\end{fulllineitems}

\index{light\_moldyn() (run.Run method)@\spxentry{light\_moldyn()}\spxextra{run.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{run:run.Run.light_moldyn}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{light\_moldyn}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
This is a function to debug a run in steps

\end{fulllineitems}

\index{check\_dist() (run.Run method)@\spxentry{check\_dist()}\spxextra{run.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{run:run.Run.check_dist}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_dist}}}{\emph{\DUrole{n}{vector1}}, \emph{\DUrole{n}{vector2}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Check distance between protein and all possible ligands (if any).
Raise warning is dist \textgreater{} 50

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Protein module}
\label{\detokenize{protein:module-protein}}\label{\detokenize{protein:protein-module}}\label{\detokenize{protein::doc}}\index{module@\spxentry{module}!protein@\spxentry{protein}}\index{protein@\spxentry{protein}!module@\spxentry{module}}
\sphinxAtStartPar
This module handles the protein and all submitted molecules around it.
\index{System (class in protein)@\spxentry{System}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.System}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{System}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{split\_system() (protein.System method)@\spxentry{split\_system()}\spxextra{protein.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.System.split_system}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{split\_system}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\index{ProteinComplex (class in protein)@\spxentry{ProteinComplex}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{ProteinComplex}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{setObjects() (protein.ProteinComplex method)@\spxentry{setObjects()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.setObjects}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setObjects}}}{\emph{\DUrole{n}{object}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets an object.

\end{fulllineitems}

\index{getObjects() (protein.ProteinComplex method)@\spxentry{getObjects()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.getObjects}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getObjects}}}{\emph{\DUrole{n}{object}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{set\_nanom() (protein.ProteinComplex method)@\spxentry{set\_nanom()}\spxextra{protein.ProteinComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.ProteinComplex.set_nanom}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_nanom}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Convert dimension measurements to nanometers for GROMACS

\end{fulllineitems}


\end{fulllineitems}

\index{Protein (class in protein)@\spxentry{Protein}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Protein}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{check\_number\_of\_chains() (protein.Protein method)@\spxentry{check\_number\_of\_chains()}\spxextra{protein.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Protein.check_number_of_chains}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_number\_of\_chains}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Determine if a PDB is a Monomer or a Dimer

\end{fulllineitems}

\index{calculate\_center() (protein.Protein method)@\spxentry{calculate\_center()}\spxextra{protein.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Protein.calculate_center}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_center}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Determine center of the coords in the self.pdb.

\end{fulllineitems}


\end{fulllineitems}

\index{Monomer (class in protein)@\spxentry{Monomer}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Monomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Monomer}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{Oligomer (class in protein)@\spxentry{Oligomer}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Oligomer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Oligomer}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Monomer}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Monomer}}}}}

\end{fulllineitems}

\index{CalculateLigandParameters (class in protein)@\spxentry{CalculateLigandParameters}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CalculateLigandParameters}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{CalculateLigandParameters}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{create\_itp() (protein.CalculateLigandParameters method)@\spxentry{create\_itp()}\spxextra{protein.CalculateLigandParameters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CalculateLigandParameters.create_itp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_itp}}}{\emph{\DUrole{n}{ligand}\DUrole{p}{:}\DUrole{w}{  }\DUrole{n}{str}}, \emph{\DUrole{n}{charge}\DUrole{p}{:}\DUrole{w}{  }\DUrole{n}{int}}}{{ $\rightarrow$ None}}
\pysigstopsignatures
\sphinxAtStartPar
Call ligpargen to create gromacs itp file and corresponding openmm
pdb file. Note that original pdb file will be replaced by opnemm pdb
file.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{pdbfile}} \textendash{} string containing local path to pdb of molecule. In commandline \sphinxhyphen{}i.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{charge}} \textendash{} interger charge of molecule. In commandline \sphinxhyphen{}c.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{numberOfOptimizations}} \textendash{} number of optimizations done by ligpargen. In cmdline \sphinxhyphen{}o.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\sphinxAtStartPar
Writes itp file and new pdf file to current dir. old pdb is saved in dir ligpargenInput. 
unneccessary ligpargen output is saved in dir ligpargenOutput.

\end{fulllineitems}

\index{add\_h() (protein.CalculateLigandParameters method)@\spxentry{add\_h()}\spxextra{protein.CalculateLigandParameters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CalculateLigandParameters.add_h}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_h}}}{\emph{\DUrole{n}{ligand}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{lpg2pmd() (protein.CalculateLigandParameters method)@\spxentry{lpg2pmd()}\spxextra{protein.CalculateLigandParameters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CalculateLigandParameters.lpg2pmd}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{lpg2pmd}}}{\emph{\DUrole{n}{cofactor}}, \emph{\DUrole{n}{index}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Converts LigParGen structure files to PyMemDyn input files.

\sphinxAtStartPar
Original files are stored as something\_backup.pdb or something\_backup.itp.

\end{fulllineitems}


\end{fulllineitems}

\index{Compound (class in protein)@\spxentry{Compound}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Compound}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
This is a super\sphinxhyphen{}class to provide common functions to added compounds
\index{check\_files() (protein.Compound method)@\spxentry{check\_files()}\spxextra{protein.Compound method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound.check_files}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_files}}}{\emph{\DUrole{o}{*}\DUrole{n}{files}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Check if files passed as {\color{red}\bfseries{}*}args exist

\end{fulllineitems}

\index{calculate\_center() (protein.Compound method)@\spxentry{calculate\_center()}\spxextra{protein.Compound method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound.calculate_center}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_center}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Determine center of the coords in the self.pdb.

\end{fulllineitems}

\index{correct\_resid() (protein.Compound method)@\spxentry{correct\_resid()}\spxextra{protein.Compound method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Compound.correct_resid}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{correct\_resid}}}{\emph{\DUrole{n}{pdb}}, \emph{\DUrole{n}{resid}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Correct the residue id to the specified residue id.

\end{fulllineitems}


\end{fulllineitems}

\index{Ligand (class in protein)@\spxentry{Ligand}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ligand}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Ligand}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{check\_forces() (protein.Ligand method)@\spxentry{check\_forces()}\spxextra{protein.Ligand method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ligand.check_forces}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check\_forces}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
A force field must give a set of forces which match every atom in
the pdb file. This showed particularly important to the ligands, as they
may vary along a very broad range of atoms

\end{fulllineitems}


\end{fulllineitems}

\index{CrystalWaters (class in protein)@\spxentry{CrystalWaters}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{CrystalWaters}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{setWaters() (protein.CrystalWaters method)@\spxentry{setWaters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.setWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setWaters}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set crystal waters

\end{fulllineitems}

\index{getWaters() (protein.CrystalWaters method)@\spxentry{getWaters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.getWaters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getWaters}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal waters

\end{fulllineitems}

\index{number (protein.CrystalWaters property)@\spxentry{number}\spxextra{protein.CrystalWaters property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.number}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{number}}}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal waters

\end{fulllineitems}

\index{count\_waters() (protein.CrystalWaters method)@\spxentry{count\_waters()}\spxextra{protein.CrystalWaters method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.CrystalWaters.count_waters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_waters}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Count and set the number of crystal waters in the pdb

\end{fulllineitems}


\end{fulllineitems}

\index{Ions (class in protein)@\spxentry{Ions}\spxextra{class in protein}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{protein.}}\sphinxbfcode{\sphinxupquote{Ions}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{protein:protein.Compound}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Compound}}}}}
\index{setIons() (protein.Ions method)@\spxentry{setIons()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.setIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setIons}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets the crystal ions

\end{fulllineitems}

\index{getIons() (protein.Ions method)@\spxentry{getIons()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.getIons}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getIons}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal ions

\end{fulllineitems}

\index{number (protein.Ions property)@\spxentry{number}\spxextra{protein.Ions property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.number}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{number}}}
\pysigstopsignatures
\sphinxAtStartPar
Get the crystal ions

\end{fulllineitems}

\index{count\_ions() (protein.Ions method)@\spxentry{count\_ions()}\spxextra{protein.Ions method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{protein:protein.Ions.count_ions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_ions}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Count and set the number of ions in the pdb

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Checks module}
\label{\detokenize{checks:module-checks}}\label{\detokenize{checks:checks-module}}\label{\detokenize{checks::doc}}\index{module@\spxentry{module}!checks@\spxentry{checks}}\index{checks@\spxentry{checks}!module@\spxentry{module}}
\sphinxAtStartPar
In this file multiple checks are defined for the protein.
\index{CheckProtein (class in checks)@\spxentry{CheckProtein}\spxextra{class in checks}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{checks:checks.CheckProtein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{checks.}}\sphinxbfcode{\sphinxupquote{CheckProtein}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{find\_missingLoops() (checks.CheckProtein method)@\spxentry{find\_missingLoops()}\spxextra{checks.CheckProtein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{checks:checks.CheckProtein.find_missingLoops}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find\_missingLoops}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check if the residue numbering is continuous, write sequence.
If loops are missing, return missingLoc.

\end{fulllineitems}

\index{find\_missingSideChains() (checks.CheckProtein method)@\spxentry{find\_missingSideChains()}\spxextra{checks.CheckProtein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{checks:checks.CheckProtein.find_missingSideChains}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{find\_missingSideChains}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check all sidechains of amino acids, save missing ones for modeller and remove 
faulty side chains from pbd.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
list of tuples in the form {[}(resID, ‘ALA’){]}

\end{description}\end{quote}

\end{fulllineitems}

\index{make\_ml\_pir() (checks.CheckProtein method)@\spxentry{make\_ml\_pir()}\spxextra{checks.CheckProtein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{checks:checks.CheckProtein.make_ml_pir}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{make\_ml\_pir}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
make\_ml\_pr: Modify missing regions and create a MODELLER alignment file (.pir)
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{work\_dir}} \textendash{} working directory

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{tgt1}} \textendash{} alignment.pir

\end{itemize}

\end{description}\end{quote}

\end{fulllineitems}

\index{refine\_protein() (checks.CheckProtein method)@\spxentry{refine\_protein()}\spxextra{checks.CheckProtein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{checks:checks.CheckProtein.refine_protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{refine\_protein}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Refine protein structure using MODELLER
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{knowns}} \textendash{} the .pdb file to be refined

\sphinxlineitem{Returns}
\sphinxAtStartPar
.pdb file of refined protein

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{aminoAcids module}
\label{\detokenize{aminoAcids:module-aminoAcids}}\label{\detokenize{aminoAcids:aminoacids-module}}\label{\detokenize{aminoAcids::doc}}\index{module@\spxentry{module}!aminoAcids@\spxentry{aminoAcids}}\index{aminoAcids@\spxentry{aminoAcids}!module@\spxentry{module}}
\sphinxAtStartPar
File for amino acid definitions
\index{AminoAcids (class in aminoAcids)@\spxentry{AminoAcids}\spxextra{class in aminoAcids}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{aminoAcids:aminoAcids.AminoAcids}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{aminoAcids.}}\sphinxbfcode{\sphinxupquote{AminoAcids}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}


\sphinxstepscope


\section{Membrane module}
\label{\detokenize{membrane:module-membrane}}\label{\detokenize{membrane:membrane-module}}\label{\detokenize{membrane::doc}}\index{module@\spxentry{module}!membrane@\spxentry{membrane}}\index{membrane@\spxentry{membrane}!module@\spxentry{module}}\index{Membrane (class in membrane)@\spxentry{Membrane}\spxextra{class in membrane}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{membrane:membrane.Membrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{membrane.}}\sphinxbfcode{\sphinxupquote{Membrane}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Set the characteristics of the membrane in the complex.
\index{set\_nanom() (membrane.Membrane method)@\spxentry{set\_nanom()}\spxextra{membrane.Membrane method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{membrane:membrane.Membrane.set_nanom}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_nanom}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Convert some measurements to nanometers to comply with GROMACS units.

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Bw4posres module}
\label{\detokenize{bw4posres:module-bw4posres}}\label{\detokenize{bw4posres:bw4posres-module}}\label{\detokenize{bw4posres::doc}}\index{module@\spxentry{module}!bw4posres@\spxentry{bw4posres}}\index{bw4posres@\spxentry{bw4posres}!module@\spxentry{module}}\begin{quote}

\sphinxAtStartPar
Date:        June 23, 2015
Email:       \sphinxhref{mailto:mauricio.esguerra@gmail.com}{mauricio.esguerra@gmail.com}

\sphinxAtStartPar
Description:
With this code we wish to do various task in one module:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Translate pdb to fasta without resorting to import Bio.

\item {} 
\sphinxAtStartPar
Align the translated fasta sequence to a Multiple Sequence Alignment (MSA)
and place Marks coming from a network of identified conserved
pair\sphinxhyphen{}distances of Venkatakrishnan et al.
clustalo \textendash{}profile1=GPCR\_inactive\_BWtags.aln \textendash{}profile2=mod1.fasta     \sphinxhyphen{}o withbwtags.aln \textendash{}outfmt=clustal \textendash{}wrap=1000 \textendash{}force \sphinxhyphen{}v \sphinxhyphen{}v \sphinxhyphen{}v

\item {} 
\sphinxAtStartPar
Translate Marks into properly identified residues in sequence. Notice that
this depends on a dictionary which uses the Ballesteros\sphinxhyphen{}Weinstein numbering.

\item {} 
\sphinxAtStartPar
From sequence ID. pull the atom\sphinxhyphen{}numbers of corresponding c\sphinxhyphen{}alphas
in the matched residues.

\end{enumerate}
\end{quote}


\bigskip\hrule\bigskip

\index{Run (class in bw4posres)@\spxentry{Run}\spxextra{class in bw4posres}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{bw4posres.}}\sphinxbfcode{\sphinxupquote{Run}}}{\emph{\DUrole{n}{pdb}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A pdb file is given as input to convert into one letter sequence
and then align to curated multiple sequence alignment and then
assign Ballesteros\sphinxhyphen{}Weinstein numbering to special positions.
\index{pdb2fas() (bw4posres.Run method)@\spxentry{pdb2fas()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.pdb2fas}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{pdb2fas}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
From pdb file convert to fasta sequence format without the use of
dependencies such as BioPython. This pdb to fasta translator
checks for the existance of c\sphinxhyphen{}alpha residues and it is
based on their 3\sphinxhyphen{}letter sequence id.

\end{fulllineitems}

\index{clustalalign() (bw4posres.Run method)@\spxentry{clustalalign()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.clustalalign}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{clustalalign}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Align the produced fasta sequence with clustalw to assign
Ballesteros\sphinxhyphen{}Weinstein marks.

\end{fulllineitems}

\index{getcalphas() (bw4posres.Run method)@\spxentry{getcalphas()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.getcalphas}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getcalphas}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Pulls out the atom numbers of c\sphinxhyphen{}alpha atoms. Restraints are
placed on c\sphinxhyphen{}alpha atoms.

\end{fulllineitems}

\index{makedisre() (bw4posres.Run method)@\spxentry{makedisre()}\spxextra{bw4posres.Run method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{bw4posres:bw4posres.Run.makedisre}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{makedisre}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a disre.itp file with atom\sphinxhyphen{}pair id’s to be restrained
using and NMR\sphinxhyphen{}style Heaviside function based on
Ballesteros\sphinxhyphen{}Weinstein tagging.

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Complex module}
\label{\detokenize{complex:module-complex}}\label{\detokenize{complex:complex-module}}\label{\detokenize{complex::doc}}\index{module@\spxentry{module}!complex@\spxentry{complex}}\index{complex@\spxentry{complex}!module@\spxentry{module}}\index{MembraneComplex (class in complex)@\spxentry{MembraneComplex}\spxextra{class in complex}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{complex.}}\sphinxbfcode{\sphinxupquote{MembraneComplex}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{setMembrane() (complex.MembraneComplex method)@\spxentry{setMembrane()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.setMembrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setMembrane}}}{\emph{\DUrole{n}{membrane}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set the membrane pdb file

\end{fulllineitems}

\index{getMembrane() (complex.MembraneComplex method)@\spxentry{getMembrane()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.getMembrane}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getMembrane}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setComplex() (complex.MembraneComplex method)@\spxentry{setComplex()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.setComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setComplex}}}{\emph{\DUrole{n}{complex}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Set the complex object

\end{fulllineitems}

\index{getComplex() (complex.MembraneComplex method)@\spxentry{getComplex()}\spxextra{complex.MembraneComplex method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{complex:complex.MembraneComplex.getComplex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{getComplex}}}{}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Queue module}
\label{\detokenize{queue:module-queue}}\label{\detokenize{queue:queue-module}}\label{\detokenize{queue::doc}}\index{module@\spxentry{module}!queue@\spxentry{queue}}\index{queue@\spxentry{queue}!module@\spxentry{module}}
\sphinxAtStartPar
A multi\sphinxhyphen{}producer, multi\sphinxhyphen{}consumer queue.
\index{Empty@\spxentry{Empty}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Empty}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Empty}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{Exception}}

\sphinxAtStartPar
Exception raised by Queue.get(block=0)/get\_nowait().

\end{fulllineitems}

\index{Full@\spxentry{Full}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Full}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Full}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{Exception}}

\sphinxAtStartPar
Exception raised by Queue.put(block=0)/put\_nowait().

\end{fulllineitems}

\index{Queue (class in queue)@\spxentry{Queue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{Queue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Create a queue object with a given maximum size.

\sphinxAtStartPar
If maxsize is \textless{}= 0, the queue size is infinite.
\index{task\_done() (queue.Queue method)@\spxentry{task\_done()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.task_done}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{task\_done}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Indicate that a formerly enqueued task is complete.

\sphinxAtStartPar
Used by Queue consumer threads.  For each get() used to fetch a task,
a subsequent call to task\_done() tells the queue that the processing
on the task is complete.

\sphinxAtStartPar
If a join() is currently blocking, it will resume when all items
have been processed (meaning that a task\_done() call was received
for every item that had been put() into the queue).

\sphinxAtStartPar
Raises a ValueError if called more times than there were items
placed in the queue.

\end{fulllineitems}

\index{join() (queue.Queue method)@\spxentry{join()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.join}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{join}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Blocks until all items in the Queue have been gotten and processed.

\sphinxAtStartPar
The count of unfinished tasks goes up whenever an item is added to the
queue. The count goes down whenever a consumer thread calls task\_done()
to indicate the item was retrieved and all work on it is complete.

\sphinxAtStartPar
When the count of unfinished tasks drops to zero, join() unblocks.

\end{fulllineitems}

\index{qsize() (queue.Queue method)@\spxentry{qsize()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.qsize}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{qsize}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return the approximate size of the queue (not reliable!).

\end{fulllineitems}

\index{empty() (queue.Queue method)@\spxentry{empty()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.empty}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{empty}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return True if the queue is empty, False otherwise (not reliable!).

\sphinxAtStartPar
This method is likely to be removed at some point.  Use qsize() == 0
as a direct substitute, but be aware that either approach risks a race
condition where a queue can grow before the result of empty() or
qsize() can be used.

\sphinxAtStartPar
To create code that needs to wait for all queued tasks to be
completed, the preferred technique is to use the join() method.

\end{fulllineitems}

\index{full() (queue.Queue method)@\spxentry{full()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.full}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{full}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Return True if the queue is full, False otherwise (not reliable!).

\sphinxAtStartPar
This method is likely to be removed at some point.  Use qsize() \textgreater{}= n
as a direct substitute, but be aware that either approach risks a race
condition where a queue can shrink before the result of full() or
qsize() can be used.

\end{fulllineitems}

\index{put() (queue.Queue method)@\spxentry{put()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.put}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put}}}{\emph{\DUrole{n}{item}}, \emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put an item into the queue.

\sphinxAtStartPar
If optional args ‘block’ is true and ‘timeout’ is None (the default),
block if necessary until a free slot is available. If ‘timeout’ is
a non\sphinxhyphen{}negative number, it blocks at most ‘timeout’ seconds and raises
the Full exception if no free slot was available within that time.
Otherwise (‘block’ is false), put an item on the queue if a free slot
is immediately available, else raise the Full exception (‘timeout’
is ignored in that case).

\end{fulllineitems}

\index{get() (queue.Queue method)@\spxentry{get()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.get}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get}}}{\emph{\DUrole{n}{block}\DUrole{o}{=}\DUrole{default_value}{True}}, \emph{\DUrole{n}{timeout}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue.

\sphinxAtStartPar
If optional args ‘block’ is true and ‘timeout’ is None (the default),
block if necessary until an item is available. If ‘timeout’ is
a non\sphinxhyphen{}negative number, it blocks at most ‘timeout’ seconds and raises
the Empty exception if no item was available within that time.
Otherwise (‘block’ is false), return an item if one is immediately
available, else raise the Empty exception (‘timeout’ is ignored
in that case).

\end{fulllineitems}

\index{put\_nowait() (queue.Queue method)@\spxentry{put\_nowait()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.put_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{put\_nowait}}}{\emph{\DUrole{n}{item}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Put an item into the queue without blocking.

\sphinxAtStartPar
Only enqueue the item if a free slot is immediately available.
Otherwise raise the Full exception.

\end{fulllineitems}

\index{get\_nowait() (queue.Queue method)@\spxentry{get\_nowait()}\spxextra{queue.Queue method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.Queue.get_nowait}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_nowait}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove and return an item from the queue without blocking.

\sphinxAtStartPar
Only get an item if one is immediately available. Otherwise
raise the Empty exception.

\end{fulllineitems}


\end{fulllineitems}

\index{PriorityQueue (class in queue)@\spxentry{PriorityQueue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.PriorityQueue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{PriorityQueue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{queue:queue.Queue}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Queue}}}}}

\sphinxAtStartPar
Variant of Queue that retrieves open entries in priority order (lowest first).

\sphinxAtStartPar
Entries are typically tuples of the form:  (priority number, data).

\end{fulllineitems}

\index{LifoQueue (class in queue)@\spxentry{LifoQueue}\spxextra{class in queue}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{queue:queue.LifoQueue}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{queue.}}\sphinxbfcode{\sphinxupquote{LifoQueue}}}{\emph{\DUrole{n}{maxsize}\DUrole{o}{=}\DUrole{default_value}{0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{queue:queue.Queue}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{Queue}}}}}

\sphinxAtStartPar
Variant of Queue that retrieves most recently added entries first.

\end{fulllineitems}


\sphinxstepscope


\section{Recipes module}
\label{\detokenize{recipes:module-recipes}}\label{\detokenize{recipes:recipes-module}}\label{\detokenize{recipes::doc}}\index{module@\spxentry{module}!recipes@\spxentry{recipes}}\index{recipes@\spxentry{recipes}!module@\spxentry{module}}
\sphinxAtStartPar
This module describes the commandline or python commands for all the 
phases of pymemdyn. It consists of:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Init

\item {} 
\sphinxAtStartPar
Minimization

\item {} 
\sphinxAtStartPar
Equilibration

\item {} 
\sphinxAtStartPar
Relaxation

\item {} 
\sphinxAtStartPar
Collecting results

\end{itemize}
\index{BasicInit (class in recipes)@\spxentry{BasicInit}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicInit}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicInit}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandInit (class in recipes)@\spxentry{LigandInit}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandInit}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandInit}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicInit}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicInit}}}}}

\end{fulllineitems}

\index{BasicMinimization (class in recipes)@\spxentry{BasicMinimization}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicMinimization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicMinimization}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicEquilibration (class in recipes)@\spxentry{BasicEquilibration}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicEquilibration}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicEquilibration}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandEquilibration (class in recipes)@\spxentry{LigandEquilibration}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandEquilibration}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandEquilibration}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicEquilibration}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicEquilibration}}}}}

\end{fulllineitems}

\index{BasicRelax (class in recipes)@\spxentry{BasicRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{LigandRelax (class in recipes)@\spxentry{LigandRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.LigandRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{LigandRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicRelax}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicRelax}}}}}

\end{fulllineitems}

\index{BasicCARelax (class in recipes)@\spxentry{BasicCARelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCARelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCARelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicBWRelax (class in recipes)@\spxentry{BasicBWRelax}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicBWRelax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicBWRelax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicCollectResults (class in recipes)@\spxentry{BasicCollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\end{fulllineitems}

\index{BasicCACollectResults (class in recipes)@\spxentry{BasicCACollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicCACollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicCACollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicCollectResults}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicCollectResults}}}}}

\end{fulllineitems}

\index{BasicBWCollectResults (class in recipes)@\spxentry{BasicBWCollectResults}\spxextra{class in recipes}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{recipes:recipes.BasicBWCollectResults}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{recipes.}}\sphinxbfcode{\sphinxupquote{BasicBWCollectResults}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{recipes:recipes.BasicCollectResults}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{BasicCollectResults}}}}}

\end{fulllineitems}


\sphinxstepscope


\section{Gromacs module}
\label{\detokenize{gromacs:module-gromacs}}\label{\detokenize{gromacs:gromacs-module}}\label{\detokenize{gromacs::doc}}\index{module@\spxentry{module}!gromacs@\spxentry{gromacs}}\index{gromacs@\spxentry{gromacs}!module@\spxentry{module}}\index{Gromacs (class in gromacs)@\spxentry{Gromacs}\spxextra{class in gromacs}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{gromacs.}}\sphinxbfcode{\sphinxupquote{Gromacs}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{set\_membrane\_complex() (gromacs.Gromacs method)@\spxentry{set\_membrane\_complex()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_membrane_complex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_membrane\_complex}}}{\emph{\DUrole{n}{value}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_membrane\_complex: Sets the membrane object

\end{fulllineitems}

\index{get\_membrane\_complex() (gromacs.Gromacs method)@\spxentry{get\_membrane\_complex()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_membrane_complex}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_membrane\_complex}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{membrane\_complex (gromacs.Gromacs property)@\spxentry{membrane\_complex}\spxextra{gromacs.Gromacs property}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.membrane_complex}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{property\DUrole{w}{  }}}\sphinxbfcode{\sphinxupquote{membrane\_complex}}}
\pysigstopsignatures
\end{fulllineitems}

\index{count\_lipids() (gromacs.Gromacs method)@\spxentry{count\_lipids()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.count_lipids}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{count\_lipids}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
count\_lipids: Counts the lipids in source and writes a target with N4 tags

\end{fulllineitems}

\index{get\_charge() (gromacs.Gromacs method)@\spxentry{get\_charge()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_charge}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_charge}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_charge: Gets the total charge of a system using gromacs grompp command

\end{fulllineitems}

\index{get\_ndx\_groups() (gromacs.Gromacs method)@\spxentry{get\_ndx\_groups()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_ndx_groups}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_ndx\_groups}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_ndx\_groups: Run make\_ndx and set the total number of groups found

\end{fulllineitems}

\index{get\_ndx\_sol() (gromacs.Gromacs method)@\spxentry{get\_ndx\_sol()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.get_ndx_sol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_ndx\_sol}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
get\_ndx\_sol: Run make\_ndx and set the last number id for SOL found

\end{fulllineitems}

\index{make\_ndx() (gromacs.Gromacs method)@\spxentry{make\_ndx()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.make_ndx}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{make\_ndx}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
make\_ndx: Wraps the make\_ndx command tweaking the input to reflect the
characteristics of the complex

\end{fulllineitems}

\index{make\_topol\_lipids() (gromacs.Gromacs method)@\spxentry{make\_topol\_lipids()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.make_topol_lipids}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{make\_topol\_lipids}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
make\_topol\_lipids: Add lipid positions to topol.top

\end{fulllineitems}

\index{manual\_log() (gromacs.Gromacs method)@\spxentry{manual\_log()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.manual_log}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{manual\_log}}}{\emph{\DUrole{n}{command}}, \emph{\DUrole{n}{output}}}{}
\pysigstopsignatures
\sphinxAtStartPar
manual\_log: Redirect the output to file in command{[}“options”{]}{[}“log”{]}
Some commands can’t be logged via flag, so one has to catch and
redirect stdout and stderr

\end{fulllineitems}

\index{relax() (gromacs.Gromacs method)@\spxentry{relax()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.relax}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{relax}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
relax: Relax a protein

\end{fulllineitems}

\index{run\_recipe() (gromacs.Gromacs method)@\spxentry{run\_recipe()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.run_recipe}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run\_recipe}}}{\emph{\DUrole{n}{debugFast}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
run\_recipe: Run recipe for the complex

\end{fulllineitems}

\index{select\_recipe() (gromacs.Gromacs method)@\spxentry{select\_recipe()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.select_recipe}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{select\_recipe}}}{\emph{\DUrole{n}{stage}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{debugFast}\DUrole{o}{=}\DUrole{default_value}{False}}}{}
\pysigstopsignatures
\sphinxAtStartPar
select\_recipe: Select the appropriate recipe for the complex

\end{fulllineitems}

\index{set\_box\_sizes() (gromacs.Gromacs method)@\spxentry{set\_box\_sizes()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_box_sizes}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_box\_sizes}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_box\_sizes: Set length values for different boxes

\end{fulllineitems}

\index{set\_chains() (gromacs.Gromacs method)@\spxentry{set\_chains()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_chains}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_chains}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_chains: Set the REAL points of a dimer after protonation

\end{fulllineitems}

\index{set\_grompp() (gromacs.Gromacs method)@\spxentry{set\_grompp()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_grompp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_grompp}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_grompp: Copy template files to working dir

\end{fulllineitems}

\index{set\_itp() (gromacs.Gromacs method)@\spxentry{set\_itp()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_itp}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_itp}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_itp: Cut a top file to be usable later as itp

\end{fulllineitems}

\index{set\_options() (gromacs.Gromacs method)@\spxentry{set\_options()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_options}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_options}}}{\emph{\DUrole{n}{options}}, \emph{\DUrole{n}{breaks}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_options: Set break options from recipe

\end{fulllineitems}

\index{set\_popc() (gromacs.Gromacs method)@\spxentry{set\_popc()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_popc}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_popc}}}{\emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_popc: Create a pdb file only with the lipid bilayer (POP), no waters.
Set some measures on the fly (height of the bilayer)

\end{fulllineitems}

\index{set\_protein\_height() (gromacs.Gromacs method)@\spxentry{set\_protein\_height()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_protein_height}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_protein\_height}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_protein\_height: Get the z\sphinxhyphen{}axis center from a pdb file for membrane or
solvent alignment

\end{fulllineitems}

\index{set\_protein\_size() (gromacs.Gromacs method)@\spxentry{set\_protein\_size()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_protein_size}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_protein\_size}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_protein\_size: Get the protein maximum base width from a pdb file

\end{fulllineitems}

\index{set\_stage\_init() (gromacs.Gromacs method)@\spxentry{set\_stage\_init()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_stage_init}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_stage\_init}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_stage\_init: Copy a set of files from source to target dir

\end{fulllineitems}

\index{set\_steep() (gromacs.Gromacs method)@\spxentry{set\_steep()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_steep}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_steep}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_steep: Copy the template steep.mdp to target dir

\end{fulllineitems}

\index{set\_water() (gromacs.Gromacs method)@\spxentry{set\_water()}\spxextra{gromacs.Gromacs method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Gromacs.set_water}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{set\_water}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
set\_water: Create a water layer for a box

\end{fulllineitems}


\end{fulllineitems}

\index{Wrapper (class in gromacs)@\spxentry{Wrapper}\spxextra{class in gromacs}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{gromacs.}}\sphinxbfcode{\sphinxupquote{Wrapper}}}{\emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{generate\_command() (gromacs.Wrapper method)@\spxentry{generate\_command()}\spxextra{gromacs.Wrapper method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper.generate_command}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{generate\_command}}}{\emph{\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
generate\_command: Receive some variables in kwargs, generate
the appropriate command to be run. Return a set in the form of
a string “command \sphinxhyphen{}with flags”

\end{fulllineitems}

\index{run\_command() (gromacs.Wrapper method)@\spxentry{run\_command()}\spxextra{gromacs.Wrapper method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{gromacs:gromacs.Wrapper.run_command}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run\_command}}}{\emph{\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
run\_command: Run a command that comes in kwargs in a subprocess, and
return the output as (output, errors)

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Groerrors module}
\label{\detokenize{groerrors:module-groerrors}}\label{\detokenize{groerrors:groerrors-module}}\label{\detokenize{groerrors::doc}}\index{module@\spxentry{module}!groerrors@\spxentry{groerrors}}\index{groerrors@\spxentry{groerrors}!module@\spxentry{module}}\index{GromacsError@\spxentry{GromacsError}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsError}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{GromacsError}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{BaseException}}

\end{fulllineitems}

\index{IOGromacsError@\spxentry{IOGromacsError}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.IOGromacsError}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{exception\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{IOGromacsError}}}{\emph{\DUrole{n}{command}}, \emph{\DUrole{n}{explain}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: {\hyperref[\detokenize{groerrors:groerrors.GromacsError}]{\sphinxcrossref{\sphinxcode{\sphinxupquote{GromacsError}}}}}

\sphinxAtStartPar
Exception raised with “File input/output error” message

\end{fulllineitems}

\index{GromacsMessages (class in groerrors)@\spxentry{GromacsMessages}\spxextra{class in groerrors}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{groerrors.}}\sphinxbfcode{\sphinxupquote{GromacsMessages}}}{\emph{\DUrole{n}{gro\_err}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{command}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{o}{*}\DUrole{n}{args}}, \emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
Load an error message and split it along as many properties as
possible
\index{e (groerrors.GromacsMessages attribute)@\spxentry{e}\spxextra{groerrors.GromacsMessages attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages.e}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{e}}\sphinxbfcode{\sphinxupquote{\DUrole{w}{  }\DUrole{p}{=}\DUrole{w}{  }\{\textquotesingle{}Can not open file\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}, \textquotesingle{}Fatal error:\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}, \textquotesingle{}File input/output error\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}, \textquotesingle{}srun: error: Unable to create job step\textquotesingle{}: \textless{}class \textquotesingle{}groerrors.IOGromacsError\textquotesingle{}\textgreater{}\}}}}
\pysigstopsignatures
\end{fulllineitems}

\index{check() (groerrors.GromacsMessages method)@\spxentry{check()}\spxextra{groerrors.GromacsMessages method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{groerrors:groerrors.GromacsMessages.check}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{check}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Check if the GROMACS error message has any of the known error
messages. Set the self.error to the value of the error

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Broker module}
\label{\detokenize{broker:module-broker}}\label{\detokenize{broker:broker-module}}\label{\detokenize{broker::doc}}\index{module@\spxentry{module}!broker@\spxentry{broker}}\index{broker@\spxentry{broker}!module@\spxentry{module}}
\sphinxAtStartPar
This is a lame broker (or message dispatcher). When Gromacs enters a run, 
it should choose a broker from here and dispatch messages through it.

\sphinxAtStartPar
Depending on the broker, the messages may be just printed or something else
\index{Printing (class in broker)@\spxentry{Printing}\spxextra{class in broker}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{broker:broker.Printing}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{broker.}}\sphinxbfcode{\sphinxupquote{Printing}}}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}
\index{dispatch() (broker.Printing method)@\spxentry{dispatch()}\spxextra{broker.Printing method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{broker:broker.Printing.dispatch}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{dispatch}}}{\emph{\DUrole{n}{msg}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Simply print the msg passed

\end{fulllineitems}


\end{fulllineitems}


\sphinxstepscope


\section{Utils module}
\label{\detokenize{utils:module-utils}}\label{\detokenize{utils:utils-module}}\label{\detokenize{utils::doc}}\index{module@\spxentry{module}!utils@\spxentry{utils}}\index{utils@\spxentry{utils}!module@\spxentry{module}}\index{clean\_pdb() (in module utils)@\spxentry{clean\_pdb()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.clean_pdb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{clean\_pdb}}}{\emph{\DUrole{n}{src}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Remove incorrectly allocated atom identifiers in pdb file

\end{fulllineitems}

\index{clean\_topol() (in module utils)@\spxentry{clean\_topol()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.clean_topol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{clean\_topol}}}{\emph{\DUrole{n}{src}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Clean the src topol of path specifics, and paste results in target

\end{fulllineitems}

\index{concat() (in module utils)@\spxentry{concat()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.concat}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{concat}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Make a whole pdb file with all the pdb provided

\end{fulllineitems}

\index{getbw() (in module utils)@\spxentry{getbw()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.getbw}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{getbw}}}{\emph{\DUrole{o}{**}\DUrole{n}{kwargs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Call the Ballesteros\sphinxhyphen{}Weistein based pair\sphinxhyphen{}distance restraint
module.

\end{fulllineitems}

\index{make\_cat() (in module utils)@\spxentry{make\_cat()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_cat}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_cat}}}{\emph{\DUrole{n}{dir1}}, \emph{\DUrole{n}{dir2}}, \emph{\DUrole{n}{name}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Very tight function to make a list of files to inject
in some GROMACS suite programs

\end{fulllineitems}

\index{make\_ffoplsaanb() (in module utils)@\spxentry{make\_ffoplsaanb()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_ffoplsaanb}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_ffoplsaanb}}}{\emph{\DUrole{n}{complex}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Join all OPLS force fields needed to run the simulation

\end{fulllineitems}

\index{make\_topol() (in module utils)@\spxentry{make\_topol()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.make_topol}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{make\_topol}}}{\emph{\DUrole{n}{template\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}/home/rkupper/apps/pymemdyn/templates\textquotesingle{}}}, \emph{\DUrole{n}{target\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{working\_dir}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\textquotesingle{}}}, \emph{\DUrole{n}{complex}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Make the topol starting from our topol.top template

\end{fulllineitems}

\index{tar\_out() (in module utils)@\spxentry{tar\_out()}\spxextra{in module utils}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{utils:utils.tar_out}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{utils.}}\sphinxbfcode{\sphinxupquote{tar\_out}}}{\emph{\DUrole{n}{src\_dir}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{tgt}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Tar everything in a src\_dir to the tar\_file

\end{fulllineitems}


\sphinxstepscope


\section{Settings module}
\label{\detokenize{settings:module-settings}}\label{\detokenize{settings:settings-module}}\label{\detokenize{settings::doc}}\index{module@\spxentry{module}!settings@\spxentry{settings}}\index{settings@\spxentry{settings}!module@\spxentry{module}}
\sphinxAtStartPar
This module handles the local settings for pymemdyn on your machine. 
The settings are mostly paths and run settings.


\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{a}
\item\relax\sphinxstyleindexentry{aminoAcids}\sphinxstyleindexpageref{aminoAcids:\detokenize{module-aminoAcids}}
\indexspace
\bigletter{b}
\item\relax\sphinxstyleindexentry{broker}\sphinxstyleindexpageref{broker:\detokenize{module-broker}}
\item\relax\sphinxstyleindexentry{bw4posres}\sphinxstyleindexpageref{bw4posres:\detokenize{module-bw4posres}}
\indexspace
\bigletter{c}
\item\relax\sphinxstyleindexentry{checks}\sphinxstyleindexpageref{checks:\detokenize{module-checks}}
\item\relax\sphinxstyleindexentry{complex}\sphinxstyleindexpageref{complex:\detokenize{module-complex}}
\indexspace
\bigletter{g}
\item\relax\sphinxstyleindexentry{groerrors}\sphinxstyleindexpageref{groerrors:\detokenize{module-groerrors}}
\item\relax\sphinxstyleindexentry{gromacs}\sphinxstyleindexpageref{gromacs:\detokenize{module-gromacs}}
\indexspace
\bigletter{m}
\item\relax\sphinxstyleindexentry{membrane}\sphinxstyleindexpageref{membrane:\detokenize{module-membrane}}
\indexspace
\bigletter{p}
\item\relax\sphinxstyleindexentry{protein}\sphinxstyleindexpageref{protein:\detokenize{module-protein}}
\indexspace
\bigletter{q}
\item\relax\sphinxstyleindexentry{queue}\sphinxstyleindexpageref{queue:\detokenize{module-queue}}
\indexspace
\bigletter{r}
\item\relax\sphinxstyleindexentry{recipes}\sphinxstyleindexpageref{recipes:\detokenize{module-recipes}}
\item\relax\sphinxstyleindexentry{run}\sphinxstyleindexpageref{run:\detokenize{module-run}}
\indexspace
\bigletter{s}
\item\relax\sphinxstyleindexentry{settings}\sphinxstyleindexpageref{settings:\detokenize{module-settings}}
\indexspace
\bigletter{u}
\item\relax\sphinxstyleindexentry{utils}\sphinxstyleindexpageref{utils:\detokenize{module-utils}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}